<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-GB" xml:lang="en-GB"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.43">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>1&nbsp; Main GWAS pipeline – GWAS Analyses utilising the CARTaGENE database</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./tools_installation.html" rel="next">
<link href="./index.html" rel="prev">
<link href="./assets/favicon.png" rel="icon" type="image/png">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-7fd0fe4245b865325b8ce8dccb604d59.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark-dc95f10abd972b9c41a78ec8992004fc.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-ffaadd7354d42997d24597a030d95797.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark-1410147302c2b15f4561c576336ac2d5.min.css" rel="prefetch" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><style>html{ scroll-behavior: smooth; }</style>
<script async="" src="https://hypothes.is/embed.js"></script><script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script><link href="site_libs/tabwid-1.1.3/tabwid.css" rel="stylesheet">
<script src="site_libs/tabwid-1.1.3/tabwid.js"></script><script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>
<meta property="og:title" content="1&nbsp; Main GWAS pipeline – GWAS Analyses utilising the CARTaGENE database">
<meta property="og:image" content="https://arnauddroitlab.github.io/cartagene-gwas/figures/global-GWAS.png">
<meta property="og:site_name" content="GWAS Analyses utilising the CARTaGENE database">
<meta property="og:image:height" content="594">
<meta property="og:image:width" content="1297">
</head>
<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./CARTaGENE-GWAS-Report.html"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Main GWAS pipeline</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">GWAS and CartaGene</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/ArnaudDroitLab/cartagene-gwas" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="./GWAS-Analyses-utilising-the-CARTaGENE-database.docx" title="Download Docx" class="quarto-navigation-tool px-1" aria-label="Download Docx"><i class="bi bi-file-word"></i></a>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./CARTaGENE-GWAS-Report.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Main GWAS pipeline</span></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Appendices</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./tools_installation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">A</span>&nbsp; <span class="chapter-title">Tools installation</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Bibliographic References</span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li>
<a href="#reproduce-the-pipeline-and-questions-to-address" id="toc-reproduce-the-pipeline-and-questions-to-address" class="nav-link active" data-scroll-target="#reproduce-the-pipeline-and-questions-to-address"><span class="header-section-number">1.1</span> Reproduce the pipeline and questions to address</a>
  <ul>
<li><a href="#two-step-pipeline" id="toc-two-step-pipeline" class="nav-link" data-scroll-target="#two-step-pipeline"><span class="header-section-number">1.1.1</span> Two-step pipeline</a></li>
  <li><a href="#reproducibility" id="toc-reproducibility" class="nav-link" data-scroll-target="#reproducibility"><span class="header-section-number">1.1.2</span> Reproducibility</a></li>
  <li><a href="#sec-questions" id="toc-sec-questions" class="nav-link" data-scroll-target="#sec-questions"><span class="header-section-number">1.1.3</span> Important Questions to Address!!</a></li>
  </ul>
</li>
  <li>
<a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction"><span class="header-section-number">1.2</span> Introduction</a>
  <ul>
<li><a href="#gwas-litterature-review" id="toc-gwas-litterature-review" class="nav-link" data-scroll-target="#gwas-litterature-review"><span class="header-section-number">1.2.1</span> GWAS Litterature Review</a></li>
  <li><a href="#original-design" id="toc-original-design" class="nav-link" data-scroll-target="#original-design"><span class="header-section-number">1.2.2</span> Original Design</a></li>
  <li><a href="#impute-unobserved-gwas-in-cartagene" id="toc-impute-unobserved-gwas-in-cartagene" class="nav-link" data-scroll-target="#impute-unobserved-gwas-in-cartagene"><span class="header-section-number">1.2.3</span> Impute unobserved GWAS in <code>CARTaGENE</code></a></li>
  <li><a href="#sample-size-calculation" id="toc-sample-size-calculation" class="nav-link" data-scroll-target="#sample-size-calculation"><span class="header-section-number">1.2.4</span> Sample Size Calculation</a></li>
  </ul>
</li>
  <li>
<a href="#analyses" id="toc-analyses" class="nav-link" data-scroll-target="#analyses"><span class="header-section-number">1.3</span> Analyses</a>
  <ul>
<li>
<a href="#setup-enviroment" id="toc-setup-enviroment" class="nav-link" data-scroll-target="#setup-enviroment"><span class="header-section-number">1.3.1</span> Setup enviroment</a>
  <ul class="collapse">
<li><a href="#bash-configuration" id="toc-bash-configuration" class="nav-link" data-scroll-target="#bash-configuration"><span class="header-section-number">1.3.1.1</span> Bash configuration:</a></li>
  <li><a href="#r-configuration" id="toc-r-configuration" class="nav-link" data-scroll-target="#r-configuration"><span class="header-section-number">1.3.1.2</span> R configuration:</a></li>
  </ul>
</li>
  <li>
<a href="#compute-pca-loadings" id="toc-compute-pca-loadings" class="nav-link" data-scroll-target="#compute-pca-loadings"><span class="header-section-number">1.3.2</span> Compute PCA <strong>loadings</strong></a>
  <ul class="collapse">
<li><a href="#sec-white-selection" id="toc-sec-white-selection" class="nav-link" data-scroll-target="#sec-white-selection"><span class="header-section-number">1.3.2.1</span> Step 1: Merge genotype arrays</a></li>
  <li><a href="#sec-preprocessing-redundancy" id="toc-sec-preprocessing-redundancy" class="nav-link" data-scroll-target="#sec-preprocessing-redundancy"><span class="header-section-number">1.3.2.2</span> Step 2: Preprocessing for Removing Correlated Features</a></li>
  <li><a href="#sec-PCA" id="toc-sec-PCA" class="nav-link" data-scroll-target="#sec-PCA"><span class="header-section-number">1.3.2.3</span> Step 3: PCA</a></li>
  </ul>
</li>
  <li>
<a href="#run-gwas-analyses" id="toc-run-gwas-analyses" class="nav-link" data-scroll-target="#run-gwas-analyses"><span class="header-section-number">1.3.3</span> Run GWAS analyses</a>
  <ul class="collapse">
<li><a href="#sec-phenotype-extraction" id="toc-sec-phenotype-extraction" class="nav-link" data-scroll-target="#sec-phenotype-extraction"><span class="header-section-number">1.3.3.1</span> Step 1: Phenotype feature extraction</a></li>
  </ul>
</li>
  <li>
<a href="#sec-variant-extraction" id="toc-sec-variant-extraction" class="nav-link" data-scroll-target="#sec-variant-extraction"><span class="header-section-number">1.3.4</span> Step 2: Variants extraction</a>
  <ul class="collapse">
<li><a href="#sec-HDAC9" id="toc-sec-HDAC9" class="nav-link" data-scroll-target="#sec-HDAC9"><span class="header-section-number">1.3.4.1</span> a) HDAC-family</a></li>
  </ul>
</li>
  <li>
<a href="#step-3-gwas-analyses" id="toc-step-3-gwas-analyses" class="nav-link" data-scroll-target="#step-3-gwas-analyses"><span class="header-section-number">1.3.5</span> Step 3: GWAS Analyses</a>
  <ul class="collapse">
<li><a href="#sec-GWAS" id="toc-sec-GWAS" class="nav-link" data-scroll-target="#sec-GWAS"><span class="header-section-number">1.3.5.1</span> i) GLM and GWAS</a></li>
  <li><a href="#ii-gwas-visualisations" id="toc-ii-gwas-visualisations" class="nav-link" data-scroll-target="#ii-gwas-visualisations"><span class="header-section-number">1.3.5.2</span> ii) GWAS Visualisations</a></li>
  <li><a href="#iii-multiple-test-correction" id="toc-iii-multiple-test-correction" class="nav-link" data-scroll-target="#iii-multiple-test-correction"><span class="header-section-number">1.3.5.3</span> iii) Multiple test correction</a></li>
  </ul>
</li>
  </ul>
</li>
  <li>
<a href="#perspectives-aka-post-gwas" id="toc-perspectives-aka-post-gwas" class="nav-link" data-scroll-target="#perspectives-aka-post-gwas"><span class="header-section-number">1.4</span> Perspectives, aka post GWAS</a>
  <ul>
<li><a href="#post-hoc-gwas" id="toc-post-hoc-gwas" class="nav-link" data-scroll-target="#post-hoc-gwas"><span class="header-section-number">1.4.1</span> Post-hoc GWAS</a></li>
  <li><a href="#explore-non-coding-regulatory-regions-from-wes-to-wgs" id="toc-explore-non-coding-regulatory-regions-from-wes-to-wgs" class="nav-link" data-scroll-target="#explore-non-coding-regulatory-regions-from-wes-to-wgs"><span class="header-section-number">1.4.2</span> Explore non-coding, regulatory regions (from WES to WGS)</a></li>
  <li><a href="#post-gwas-explore-other-types-of-genetic-variations" id="toc-post-gwas-explore-other-types-of-genetic-variations" class="nav-link" data-scroll-target="#post-gwas-explore-other-types-of-genetic-variations"><span class="header-section-number">1.4.3</span> Post-GWAS: Explore other types of genetic variations</a></li>
  <li><a href="#sec-post-GWAS" id="toc-sec-post-GWAS" class="nav-link" data-scroll-target="#sec-post-GWAS"><span class="header-section-number">1.4.4</span> Post-GWAS: Retrieve automatically variants of interest</a></li>
  <li><a href="#post-gwas-incorporate-familial-pedigrees" id="toc-post-gwas-incorporate-familial-pedigrees" class="nav-link" data-scroll-target="#post-gwas-incorporate-familial-pedigrees"><span class="header-section-number">1.4.5</span> Post-GWAS: Incorporate familial pedigrees</a></li>
  <li><a href="#post-gwas-polygenic-risk-scores" id="toc-post-gwas-polygenic-risk-scores" class="nav-link" data-scroll-target="#post-gwas-polygenic-risk-scores"><span class="header-section-number">1.4.6</span> Post-GWAS: Polygenic risk scores</a></li>
  <li>
<a href="#post-gwas-orthogonal-integration" id="toc-post-gwas-orthogonal-integration" class="nav-link" data-scroll-target="#post-gwas-orthogonal-integration"><span class="header-section-number">1.4.7</span> Post-GWAS: Orthogonal integration</a>
  <ul class="collapse">
<li><a href="#epigenomic" id="toc-epigenomic" class="nav-link" data-scroll-target="#epigenomic"><span class="header-section-number">1.4.7.1</span> Epigenomic</a></li>
  <li><a href="#eqtl-quantitative-expression" id="toc-eqtl-quantitative-expression" class="nav-link" data-scroll-target="#eqtl-quantitative-expression"><span class="header-section-number">1.4.7.2</span> eQTL (quantitative expression)</a></li>
  </ul>
</li>
  </ul>
</li>
  <li>
<a href="#appendix" id="toc-appendix" class="nav-link" data-scroll-target="#appendix"><span class="header-section-number">1.5</span> Appendix</a>
  <ul>
<li><a href="#sec-DSL-Nextflow" id="toc-sec-DSL-Nextflow" class="nav-link" data-scroll-target="#sec-DSL-Nextflow"><span class="header-section-number">1.5.1</span> Scalability with <code>Nextflow</code></a></li>
  </ul>
</li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/ArnaudDroitLab/cartagene-gwas/edit/master/CARTaGENE-GWAS-Report.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ArnaudDroitLab/cartagene-gwas/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/ArnaudDroitLab/cartagene-gwas/blob/master/CARTaGENE-GWAS-Report.qmd" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content column-body" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<h1 class="title">
<span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Main GWAS pipeline</span>
</h1>
</div>



<div class="quarto-title-meta column-body">

    
  
    
  </div>
  


</header><section id="reproduce-the-pipeline-and-questions-to-address" class="level2" data-number="1.1"><h2 data-number="1.1" class="anchored" data-anchor-id="reproduce-the-pipeline-and-questions-to-address">
<span class="header-section-number">1.1</span> Reproduce the pipeline and questions to address</h2>
<section id="two-step-pipeline" class="level3" data-number="1.1.1"><h3 data-number="1.1.1" class="anchored" data-anchor-id="two-step-pipeline">
<span class="header-section-number">1.1.1</span> Two-step pipeline</h3>
<p>Assuming PCA components have already been computed (report to <a href="#sec-white-selection" class="quarto-xref"><span>Section 1.3.2.1</span></a> and <a href="#sec-preprocessing-redundancy" class="quarto-xref"><span>Section 1.3.2.2</span></a> for details), follow this two-step protocol:</p>
<ol type="i">
<li>Retrieve all known variants associated with your genes of interest in <a href="#sec-variant-extraction" class="quarto-xref"><span>Section 1.3.4</span></a>, and generate the corresponding <code>VCF</code> file.</li>
<li>Change the response variable to predict in <a href="#lst-glm-bash" class="quarto-xref">Listing&nbsp;<span>1.6</span></a> (with option <code>--glm</code>, the model applied should adjust automatically whether the outcome is binary or continuous).</li>
</ol></section><section id="reproducibility" class="level3" data-number="1.1.2"><h3 data-number="1.1.2" class="anchored" data-anchor-id="reproducibility">
<span class="header-section-number">1.1.2</span> Reproducibility</h3>
<p>Two command-lines tools are needed for reproducing the analyses and/or delivering new outcomes:</p>
<ul>
<li>
<a href="https://s3.amazonaws.com/plink2-assets/alpha6/plink2_linux_x86_64_20250129.zip"><code>plink2</code></a> for running most of the SNPs analyses.</li>
<li>
<a href="https://github.com/samtools/bcftools/releases/download/1.21/bcftools-1.21.tar.bz2"><code>bcftools</code></a> for extracting variants of interest in a given genomic region.</li>
<li>Details for installing from scratch these two CLI tools are reported in <a href="tools_installation.html#sec-tools-install" class="quarto-xref"><span>Section A.1</span></a>.</li>
</ul>
<p>The code, main results (figures and tables), along with reports, are versioned in an unique place, on <code>GitHub</code>:</p>
<ul>
<li>
<a href="https://github.com/ArnaudDroitLab/cartagene-gwas"><code>GitHub</code> project: <code>cartagene-gwas</code></a> to version the code</li>
<li>R project + <code>renv()</code> to create a reproducible and standalone computational environment.</li>
</ul></section><section id="sec-questions" class="level3" data-number="1.1.3"><h3 data-number="1.1.3" class="anchored" data-anchor-id="sec-questions">
<span class="header-section-number">1.1.3</span> Important Questions to Address!!</h3>
<ul>
<li><span data-fg="red" style="color: red;">Check Human reference genome assembly, see <a href="#sec-HDAC9" class="quarto-xref"><span>Section 1.3.4.1</span></a>. Indeed, among the five genome arrays assembled in <a href="#sec-white-selection" class="quarto-xref"><span>Section 1.3.2.1</span></a>, most have been mapped against <code>GRCh37 (hg19)</code>, but at least one has been mapped against the recent <code>hg38 (GRCh38, 2013)</code> version, report to page 38 in <span class="citation" data-cites="pelletier2022">Pelletier (<a href="references.html#ref-pelletier2022" role="doc-biblioref">2022</a>)</span>.</span></li>
<li>We definitely do not have the most recent version of <code>CarTaGene</code> (in the paper, around <span class="math inline">\(30000\)</span> samples mentioned, against <span class="math inline">\(20000\)</span> in my database + )</li>
<li><span data-fg="red" style="color: red;">Why starting from BAM/BED/BIM pre-processed files in <a href="#sec-white-selection" class="quarto-xref"><span>Section 1.3.2.1</span></a>, instead of <code>FASTQ</code> files available in <code>/mnt/projects_tn01/Cartagene/data/merged</code>? Related question: we use the <code>BAM/BED/BIM</code> files stored in folder <code>/mnt/projects_tn01/Cartagene/analyses/QC</code> for PCA computation in <a href="#sec-preprocessing-redundancy" class="quarto-xref"><span>Section 1.3.2.2</span></a>, but <code>VCF</code> files stored in <code>/mnt/projects_tn01/Cartagene/analyses/variants_extraction</code> for running the GWAS analyses, for which reason?</span></li>
<li>
<span data-fg="red" style="color: red;">Why use a biased <em>gene-centric approach</em>, instead of true GWAS (further discussed in <a href="#sec-post-GWAS" class="quarto-xref"><span>Section 1.4.4</span></a>)?</span>. Specifically, in contrast with the conclusions of this limited study on a single gene, namely <code>HDAC-9</code>, running a whole GWAS analysis on <code>CarTaGene</code> revealed promising and significant SNPs, as illustrated in <a href="#fig-gwas-global" class="quarto-xref">Figure&nbsp;<span>1.1</span></a>.
<ul>
<li>An alternative to <em>candidate gene</em> association studies relies on the <em>linkage analysis approach</em>, which uses familial pedigrees to map genetic variants underlying common human diseases. <strong>This strategy requires both genetic data and detailed family pedigrees.</strong>, report to for details.</li>
</ul>
</li>
<li>Address intrinsic heterogeneity of the sampled population, see <a href="https://doi.org/10.1093/bib/bbad437">A benchmark study on current GWAS models in admixed populations</a>, from <span class="citation" data-cites="yang2024bib">Yang et al. (<a href="references.html#ref-yang2024bib" role="doc-biblioref">2024</a>)</span>.</li>
</ul>
<div id="fig-gwas-global" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-gwas-global-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="./figures/global-GWAS.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Figure&nbsp;1.1: GWAS of CarTaGene SNPs using TOPMed Merge-Impute (first, merge all genotype arrays, then impute SNPs) strategy. obtained through genotyping and imputation using the TOPMed Imputation Server under two analytic workflows: Impute-Merge and Merge-Impute. Manhattan plots depict the genome-wide association study (GWAS) results for variants (Single nucleotide polymorphisms and insertion-deletion), before and after imputation. The y-axis represents the –\log_{10}(p) for the association of each variant. Variants with p-values equal to zero are highlighted in green, while the horizontal red line indicates the genome-wide significance threshold.. The total sample consisted of 27,422 individuals."><img src="./figures/global-GWAS.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-gwas-global-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1.1: <strong>GWAS of <code>CarTaGene</code> SNPs using <code>TOPMed</code> Merge-Impute (first, merge all genotype arrays, then impute SNPs) strategy</strong>. obtained through genotyping and imputation using the <code>TOPMed</code> Imputation Server under two analytic workflows: Impute-Merge and Merge-Impute. Manhattan plots depict the genome-wide association study (GWAS) results for variants (<em>Single nucleotide polymorphisms</em> and <em>insertion-deletion</em>), before and after imputation. The <span class="math inline">\(y\)</span>-axis represents the <span class="math inline">\(–\log_{10}(p)\)</span> for the association of each variant. <span data-fg="green" style="color: green;">Variants with <span class="math inline">\(p\)</span>-values equal to zero are highlighted in green</span>, while the <span data-fg="red" style="color: red;">horizontal red line indicates the genome-wide <em>significance threshold</em>.</span>. The total sample consisted of <span class="math inline">\(27,422\)</span> individuals.
</figcaption></figure>
</div>
</section></section><section id="introduction" class="level2" data-number="1.2"><h2 data-number="1.2" class="anchored" data-anchor-id="introduction">
<span class="header-section-number">1.2</span> Introduction</h2>
<section id="gwas-litterature-review" class="level3" data-number="1.2.1"><h3 data-number="1.2.1" class="anchored" data-anchor-id="gwas-litterature-review">
<span class="header-section-number">1.2.1</span> GWAS Litterature Review</h3>
<ul>
<li><p>Fundamental paper, aka <a href="https://www.nature.com/articles/nrg1522">Genome-wide association studies: theoretical and practical concerns</a>, from <span class="citation" data-cites="wang2005nrg">Wang et al. (<a href="references.html#ref-wang2005nrg" role="doc-biblioref">2005</a>)</span>.</p></li>
<li><p>Book reviewing pros and cons of statistical GWAS approaches, from <a href="https://doi.org/10.1007/978-1-62703-447-0_6">Overview of Statistical Methods for Genome-Wide Association Studies</a>, from <span class="citation" data-cites="hayes2013gasagp">Hayes (<a href="references.html#ref-hayes2013gasagp" role="doc-biblioref">2013</a>)</span>.</p></li>
<li><p><a href="https://www.nature.com/articles/s43586-021-00056-9">Genome-wide association studies</a>, from <span class="citation" data-cites="uffelmann2021nrmp">Uffelmann et al. (<a href="references.html#ref-uffelmann2021nrmp" role="doc-biblioref">2021</a>)</span>.</p></li>
</ul></section><section id="original-design" class="level3" data-number="1.2.2"><h3 data-number="1.2.2" class="anchored" data-anchor-id="original-design">
<span class="header-section-number">1.2.2</span> Original Design</h3>
<ul>
<li>
<a href="https://doi.org/10.1093/ije/dys160">Cohort profile of the <code>CARTaGENE</code> study: Quebec’s population-based biobank for public health and personalized genomics</a>, from <span class="citation" data-cites="awadalla2013ijoe">Awadalla et al. (<a href="references.html#ref-awadalla2013ijoe" role="doc-biblioref">2013</a>)</span>.</li>
</ul></section><section id="impute-unobserved-gwas-in-cartagene" class="level3" data-number="1.2.3"><h3 data-number="1.2.3" class="anchored" data-anchor-id="impute-unobserved-gwas-in-cartagene">
<span class="header-section-number">1.2.3</span> Impute unobserved GWAS in <code>CARTaGENE</code>
</h3>
<ul>
<li><p>Most of the <code>CARTaGENE</code> samples are <em>WES</em> (whole exome sequencing), instead of most complete <em>WGS</em> (whole genome sequencing).</p></li>
<li>
<p><a href="https://umontreal.scholaris.ca/bitstreams/83d64aa1-da20-4a6f-8ae0-9dcb0165417d/download">Évaluation de l’imputation des données génétiques Canadiennes-Françaises</a>, from <span class="citation" data-cites="pelletier2022">Pelletier (<a href="references.html#ref-pelletier2022" role="doc-biblioref">2022</a>)</span>. <strong>Key points</strong>:</p>
<ul>
<li>
<strong>Introduction section</strong>: presents the most critical issues when dealing with GWAS information: linkage disequilibrium,</li>
<li>Paper, corresponding to Chapter 3, named “Evaluation of genetic imputation in the French-Canadian founder population”, pages 54-93. Details the underlying imputation strategy, derived from <code>TOPMed</code> (for Trans-Omics for Precision Medicine).</li>
<li>
<strong>Page 61</strong>: Describes the imputation strategy, along with the preprocessing steps. Code for reproducing the analyses is <a href="https://github.com/JustinPelletier/FC-imputation">FC-imputation</a>, kinda messy I must admit. <strong>Imputation performance</strong> is much lower for SNPs exhibiting <code>MAF&lt;0.01</code>, strongly promoting early discarding.<br>
</li>
<li>
<strong>Page 99</strong>: Strong polymorphism in some regions, such as <code>HLA</code>.</li>
<li>Reach out <a href="mailto:julie.hussin@umontreal.ca">Julie Hussin</a>, or <a href="https://www.linkedin.com/in/hussinju/">LinkedIn profile</a>, who actually ran the PCA computations and global GWAS analyses.]{fg=“red”}</li>
</ul>
</li>
</ul></section><section id="sample-size-calculation" class="level3" data-number="1.2.4"><h3 data-number="1.2.4" class="anchored" data-anchor-id="sample-size-calculation">
<span class="header-section-number">1.2.4</span> Sample Size Calculation</h3>
<ul>
<li><span data-fg="red" style="color: red;">Should be carried out prior to collecting GWAS samples!!</span></li>
<li>
<a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC9863799/#sec5-life-13-00235">Sample Size Calculation in Genetic Association Studies: A Practical Approach</a>, from <span class="citation" data-cites="politi2023l">Politi et al. (<a href="references.html#ref-politi2023l" role="doc-biblioref">2023</a>)</span>, notably listing the following tools:
<ul>
<li>
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7666027/"><code>GENPWR</code>: Power and Sample Size Calculations for Genetic Association Studies in the Presence of Genetic Model Mis-Specification</a>, from <span class="citation" data-cites="moore2019hh">Moore, Jacobso, and Fingerlin (<a href="references.html#ref-moore2019hh" role="doc-biblioref">2019</a>)</span> and avalaible as a <a href="https://cran.r-project.org/web/packages/genpwr/vignettes/vignette.html">R package</a>, is able to compute the sample size for dichotomous or continuous outcomes in gene-only studies, implementing additive, dominant and recessive models (see <a href="#tip-genetic-model" class="quarto-xref">Tip&nbsp;<span>1.1</span></a> for details).</li>
<li>
<a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC2387159/"><code>PGA</code>: Power for Genetic Association</a>, from <span class="citation" data-cites="menashe2008bg">Menashe, Rosenberg, and Chen (<a href="references.html#ref-menashe2008bg" role="doc-biblioref">2008</a>)</span> is able to adjust for <strong>multiple comparisons</strong>, in contrast with <code>GENPWR</code>, and LD between the marker and the disease allele. <span data-fg="red" style="color: red;">Seems not avalaible anymore</span>
</li>
</ul>
</li>
</ul>
<div id="tip-genetic-model" class="callout callout-style-default callout-tip callout-titled" title="Genetic models">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Tip&nbsp;1.1: Genetic models
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>
<p><strong>Genotype</strong>: At a given SNP with alleles <strong>A</strong> (major) and <strong>a</strong> (minor), the genotypes are:</p>
<ul>
<li>
<strong>AA</strong> (homozygous <em>major</em>)</li>
<li>
<strong>Aa</strong> (<em>heterozygous</em>)</li>
<li>
<strong>aa</strong> (homozygous <em>minor</em>)</li>
</ul>
</li>
<li><p><strong>Phenotype</strong>: A continuous or binary outcome (e.g., height, disease status)</p></li>
<li><p><strong>Notations</strong>: <span class="math inline">\(Y\)</span>: phenotype, <span class="math inline">\(G\)</span>: genotype, <span class="math inline">\(\beta\)</span>: effect size, <span class="math inline">\(X\)</span>: covariates (e.g., age, sex, PCs) and <span class="math inline">\(\epsilon\)</span>: error term.</p></li>
</ul>
<hr>
<p>🧬 Genetic Models in GWAS</p>
<ul>
<li><p><strong>Additive model</strong>: in <a href="#eq-genetic-model" class="quarto-xref">Equation&nbsp;<span>1.1</span></a>, ‘gene cat’ is converted as <code>AA = 0</code>, <code>Aa = 1</code> and <code>aa = 2</code>. Each additional copy of the minor allele increases the trait linearly.</p></li>
<li><p><strong>Dominant model</strong>: in <a href="#eq-genetic-model" class="quarto-xref">Equation&nbsp;<span>1.1</span></a>, ‘gene cat’ is converted as <code>AA = 0</code>, <code>Aa = 1</code> and <code>aa = 1</code>. Presence of <strong>at least one</strong> minor allele confers full effect (no dose response).</p></li>
<li><p><strong>Recessive model</strong>: in <a href="#eq-genetic-model" class="quarto-xref">Equation&nbsp;<span>1.1</span></a>, ‘gene cat’ is converted as <code>AA = 0</code>, <code>Aa = 0</code> and <code>aa = 1</code>. Only <strong>homozygous minor</strong> genotype has an effect.</p></li>
</ul>
<p><span id="eq-genetic-model"><span class="math display">\[
Y = \beta_0 + \beta_1 G_{\text{gene cat}} + \beta_2 X + \epsilon
\tag{1.1}\]</span></span></p>
<hr>
<table class="table">
<caption>Main Genetic Models. In conclusion, use <strong>additive</strong> by default (most comprehensive), and explore <strong>dominant/recessive</strong> in <strong>post-GWAS</strong> or when you have strong biological priors (e.g., loss-of-function){#tbl-genetic-model}</caption>
<colgroup>
<col style="width: 8%">
<col style="width: 11%">
<col style="width: 22%">
<col style="width: 28%">
<col style="width: 28%">
</colgroup>
<thead><tr class="header">
<th>Model</th>
<th>Coding</th>
<th>Assumes effect from</th>
<th>Pros</th>
<th>Cons</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Additive</td>
<td>0 / 1 / 2</td>
<td>Per additional allele</td>
<td>High power, interpretable</td>
<td>May miss non-linear effects</td>
</tr>
<tr class="even">
<td>Dominant</td>
<td>0 / 1 / 1</td>
<td>Any copy of minor allele</td>
<td>Fits some Mendelian patterns</td>
<td>Oversimplifies if dose-response</td>
</tr>
<tr class="odd">
<td>Recessive</td>
<td>0 / 0 / 1</td>
<td>Two copies of minor allele</td>
<td>Fits true recessive traits</td>
<td>Low power with rare alleles</td>
</tr>
</tbody>
</table>
<hr>
<p>Other less Frequent Models include <strong>Overdominant</strong>, in which the effect only occurs in <strong>heterozygotes</strong>. Rarely used in GWAS, but relevant for <strong>heterozygote advantage</strong> scenarios (e.g., sickle cell and malaria).</p>
</div>
</div>
</section></section><section id="analyses" class="level2" data-number="1.3"><h2 data-number="1.3" class="anchored" data-anchor-id="analyses">
<span class="header-section-number">1.3</span> Analyses</h2>
<section id="setup-enviroment" class="level3" data-number="1.3.1"><h3 data-number="1.3.1" class="anchored" data-anchor-id="setup-enviroment">
<span class="header-section-number">1.3.1</span> Setup enviroment</h3>
<section id="bash-configuration" class="level4" data-number="1.3.1.1"><h4 data-number="1.3.1.1" class="anchored" data-anchor-id="bash-configuration">
<span class="header-section-number">1.3.1.1</span> Bash configuration:</h4>
<p>We use <em>symbolic links</em> to encapsulate the whole project, see <a href="https://www.linkedin.com/posts/%F0%9F%8E%AF-ming-tommy-tang-40650014_the-most-underrated-unix-command-for-bioinformatics-activity-7287842704753848321-qhe"><code>ln -s</code> Post</a> for details:</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">## Render executive files findable</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">PATH</span><span class="op">=</span>./bin/bcftools/bin:<span class="va">$PATH</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">PATH</span><span class="op">=</span>./bin/plink2:<span class="va">$PATH</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">## Check versions</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="ex">plink2</span> <span class="at">--version</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"\n"</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="ex">bcftools</span> <span class="at">--version</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co">## Create symbolic links to organise everythin within the same folder</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co">## warning: if original content is deleted, everything broke!!!!!!</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> ln <span class="at">-s</span> ./data/genotypages/ /mnt/projects_tn01/Cartagene/genotypage/imputation/imputation_merged</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre class="code_output"><code>##  PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)
##  
##  
##  bcftools 1.21
##  Using htslib 1.21
##  Copyright (C) 2024 Genome Research Ltd.
##  License Expat: The MIT/Expat license
##  This is free software: you are free to change and redistribute it.
##  There is NO WARRANTY, to the extent permitted by law.
##  ln -s ./data/genotypages/ /mnt/projects_tn01/Cartagene/genotypage/imputation/imputation_merged</code></pre>
</div>
</div>
</section><section id="r-configuration" class="level4" data-number="1.3.1.2"><h4 data-number="1.3.1.2" class="anchored" data-anchor-id="r-configuration">
<span class="header-section-number">1.3.1.2</span> R configuration:</h4>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">## data wrangling and visualisations</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://haven.tidyverse.org">haven</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ardata-fr.github.io/flextable-book/">flextable</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></span>
<span><span class="co">## Required for code linking</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://downlit.r-lib.org/">downlit</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://xml2.r-lib.org">xml2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Avoid warning "Replace previous import ‘utils::findMatches’ by ‘S4Vectors::findMatches’ when loading ‘AnnotationDbi’"</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://conflicted.r-lib.org/">conflicted</a></span><span class="op">)</span></span>
<span><span class="fu">conflicted</span><span class="fu">::</span><span class="fu"><a href="https://conflicted.r-lib.org/reference/conflict_prefer.html">conflict_prefer</a></span><span class="op">(</span><span class="st">"findMatches"</span>, <span class="st">"S4Vectors"</span><span class="op">)</span></span>
<span><span class="fu">conflicted</span><span class="fu">::</span><span class="fu"><a href="https://conflicted.r-lib.org/reference/conflicts_prefer.html">conflicts_prefer</a></span><span class="op">(</span><span class="fu">base</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/r/base/sets.html">setdiff</a></span>, <span class="fu">base</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span>, </span>
<span>                             <span class="fu">base</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/r/base/sets.html">setequal</a></span>, <span class="fu">base</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/r/base/sets.html">union</a></span><span class="op">)</span></span>
<span><span class="fu">conflicted</span><span class="fu">::</span><span class="fu"><a href="https://conflicted.r-lib.org/reference/conflicts_prefer.html">conflicts_prefer</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="va"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>, <span class="fu">dplyr</span><span class="fu">::</span><span class="va"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co">## Retrieve gene positions</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span>  <span class="co">## Provides gene symbol to Entrez ID mapping</span></span>
<span></span>
<span><span class="co">## for generating nice visualisations</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="st">"R/gwas_plots.R"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section></section><section id="compute-pca-loadings" class="level3" data-number="1.3.2"><h3 data-number="1.3.2" class="anchored" data-anchor-id="compute-pca-loadings">
<span class="header-section-number">1.3.2</span> Compute PCA <strong>loadings</strong>
</h3>
<section id="sec-white-selection" class="level4" data-number="1.3.2.1"><h4 data-number="1.3.2.1" class="anchored" data-anchor-id="sec-white-selection">
<span class="header-section-number">1.3.2.1</span> Step 1: Merge genotype arrays</h4>
<p>Genotypes were split into five different genotyping arrays. Arrays were merged together using Bash script <a href="./mnt/projects_tn01/Cartagene/analyses/QC/merge_datasets.sh">merge_datasets.sh</a>, and command <code>--bmerge</code> of tool <code>plink2</code>. This command only keeps matching <strong>SNPs</strong> and <strong>alleles</strong> across the 5 arrays.</p>
<p>The resulting <code>BAM</code> files (sequence alignments), <code>BED</code> files (genomic regions of interest, for viewers), <code>BIM</code> (SNP information) and <code>FAM</code> (phenotype data, such as individuals and pedigree), are listed in folder <code>/mnt/projects_tn01/Cartagene/analyses/QC</code>, with prefix <code>merge_5_*</code> (for 5 genotypes concatenated).</p>
</section><section id="sec-preprocessing-redundancy" class="level4" data-number="1.3.2.2"><h4 data-number="1.3.2.2" class="anchored" data-anchor-id="sec-preprocessing-redundancy">
<span class="header-section-number">1.3.2.2</span> Step 2: Preprocessing for Removing Correlated Features</h4>
<p><strong>Remark</strong>: the prefix <code>eur_only</code> stands for Caucasian phenotypes.</p>
<section id="sec-missing-SNPs" class="level5" data-number="1.3.2.2.1"><h5 data-number="1.3.2.2.1" class="anchored" data-anchor-id="sec-missing-SNPs">
<span class="header-section-number">1.3.2.2.1</span> i) Trim missing SNPs</h5>
<ul>
<li>
<p><strong>Inputs</strong>:</p>
<ul>
<li>
<strong>Phenotype Ids of white individuals</strong>: <code>eur_only/cartagene_self_reported_EUR.plink_format.txt</code>
</li>
<li>
<strong>BIM/BED/FAM/hh</strong> folder generated with <code>--make-bed</code> command: <code>analyses/QC/merge_5_datasets</code>. <span data-fg="red" style="color: red;">It seems that among the 5 genotype arrays concatenated, at least one genotype, namely <code>gsa.17k.final.hg19.bim,</code> has been mapped thanks to <code>GRCh37 (hg19)</code> reference instead of more recent <code>hg38 (GRCh38, 2013)</code> version.</span>$$</li>
</ul>
</li>
<li><p><strong>Objective</strong>: Keep white individuals and remove SNPs variants with genotyping rate <span class="math inline">\(&lt;0.95\)</span>. The <strong>genotyping rate</strong> measures the proportion of successfully genotyped markers and indicates how complete the genotype data is. Removing SNPs with low Genotyping Rate increases Missing Data imputation performance, and overall maintains higher statistical power.</p></li>
<li><p><span data-fg="red" style="color: red;"><strong>Remark</strong>: Also advised to exclude samples with low Genotyping Rate, below 0.98.</span></p></li>
<li><p><strong>Bash command</strong>:</p></li>
</ul>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">plink2</span> <span class="dt">\</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--bfile</span> /mnt/projects_tn01/Cartagene/analyses/QC/merge_5_datasets <span class="dt">\</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--geno</span> 0.05 <span class="dt">\</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--keep</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/cartagene_self_reported_EUR.plink_format.txt <span class="dt">\</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--make-bed</span> <span class="dt">\</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">--out</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/merge_5_datasets.eur_only.geno095</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section><section id="sec-HWD" class="level5" data-number="1.3.2.2.2"><h5 data-number="1.3.2.2.2" class="anchored" data-anchor-id="sec-HWD">
<span class="header-section-number">1.3.2.2.2</span> ii) Hardy-Weinberg Disequilibrium</h5>
<ul>
<li>
<p><strong>Inputs</strong>:</p>
<ul>
<li>
<strong>BIM/BED/FAM/hh (Homozygous-Haplotype)</strong> files generated from previous SNP missing removal, see <a href="#sec-missing-SNPs" class="quarto-xref"><span>Section 1.3.2.2.1</span></a>.</li>
</ul>
</li>
<li><p><strong>Objective</strong>: Remove SNPs variants with <strong>Hardy-Weinberg disequilibrium</strong> <span class="math inline">\(&lt; 1 \times 10^{-6}\)</span>. Indeed, if a SNP is in HWD, it might reflect hidden sub-population structure or poor-quality rather than true genetic associations.</p></li>
<li><p><span data-fg="red" style="color: red;"><strong>Remark</strong>: If a SNP is under strong <strong>natural selection</strong>, such as SNPs involved in the HLA genes, they are likely to deviate from HWE due to balancing selection. If the SNP is biologically important, don’t exclude it blindly!! Alternative: compute the SNP score.</span></p></li>
<li><p><strong>Bash command</strong>:</p></li>
</ul>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">plink2</span> <span class="dt">\</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--bfile</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/merge_5_datasets.eur_only.geno095 <span class="dt">\</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--hwe</span> 1e-6 <span class="dt">\</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--make-bed</span> <span class="dt">\</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--out</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/merge_5_datasets.eur_only.geno095.hwe1e06</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section><section id="sec-LD" class="level5" data-number="1.3.2.2.3"><h5 data-number="1.3.2.2.3" class="anchored" data-anchor-id="sec-LD">
<span class="header-section-number">1.3.2.2.3</span> iii) Linkage Disequilibrium</h5>
<section id="sec-local-LD" class="level6" data-number="1.3.2.2.3.1"><h6 data-number="1.3.2.2.3.1" class="anchored" data-anchor-id="sec-local-LD">
<span class="header-section-number">1.3.2.2.3.1</span> Local LD analysis</h6>
<ul>
<li>
<p><strong>Inputs</strong>:</p>
<ul>
<li>
<strong>BIM/BED/FAM/hh (Homozygous-Haplotype)</strong> files <code>/merge_5_datasets.eur_only.geno095.hwe1e0</code> generated from previous HWD SNP trimming, see <a href="#sec-HWD" class="quarto-xref"><span>Section 1.3.2.2.2</span></a>.</li>
</ul>
</li>
<li><p><strong>Objective</strong>: Remove SNPs variants associated with strong <strong>linkage disequilibrium</strong>. In an ideal population under random mating, allele combinations should be independent (<span class="math inline">\(r^2=0\)</span>). However, due to factors such as genetic drift, or physical proximity, certain alleles tend to be inherited together more often than expected. In GWAS, LD pruning avoids <strong>overfitting</strong> (SNPs in high LD carry redundant information), avoiding spurious inflation of GWAS association signals.</p></li>
<li><p><span data-fg="red" style="color: red;"><strong>Remark</strong>: Report the Number of SNPs Before and After Pruning, usually more stringent <span class="math inline">\(R^2\)</span> is considered.</span></p></li>
<li><p><strong>Tool</strong>: <code>plink --indep-pairwise 50 5 0.5</code> will discard SNPs with a correlation coefficient above <span class="math inline">\(0.5\)</span> (given that a <span class="math inline">\(R^2\)</span> score of 1 indicates a perfect correlation), on a <em>rolling window</em> of <span class="math inline">\(50\)</span> SNPs (focus on local zones) and step <span class="math inline">\(5\)</span> by <span class="math inline">\(5\)</span>.</p></li>
<li><p><strong>Bash command</strong>:</p></li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">## identify SNPs with low LD</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="ex">plink2</span> <span class="dt">\</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--bfile</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/merge_5_datasets.eur_only.geno095.hwe1e06 <span class="dt">\</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--indep-pairwise</span> 50 5 0.5 <span class="dt">\</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--out</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/merge_5_datasets.eur_only.geno095.hwe1e06.pruned_snps</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co">## save pruned SNPs associated with low LD</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="ex">plink2</span> <span class="dt">\</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">--bfile</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/merge_5_datasets.eur_only.geno095.hwe1e06 <span class="dt">\</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">--extract</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/merge_5_datasets.eur_only.geno095.hwe1e06.pruned_snps.prune.in <span class="dt">\</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">--make-bed</span> <span class="dt">\</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">--out</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/merge_5_datasets.eur_only.geno095.hwe1e06.pruned_snps_bfile</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section><section id="sec-global-LD" class="level6" data-number="1.3.2.2.3.2"><h6 data-number="1.3.2.2.3.2" class="anchored" data-anchor-id="sec-global-LD">
<span class="header-section-number">1.3.2.2.3.2</span> Global LD analysis from prior expert knowledge</h6>
<ul>
<li>
<p><strong>Inputs</strong>:</p>
<ul>
<li>
<strong>BIM/BED/FAM/hh (Homozygous-Haplotype)</strong> files <code>merge_5_datasets.eur_only.geno095.hwe1e06.pruned_snps_bfile</code> generated from previous local LD trimming, see <a href="#sec-local-LD" class="quarto-xref"><span>Section 1.3.2.2.3.1</span></a>.</li>
</ul>
</li>
<li><p><strong>Objective</strong>: Remove SNPs variants associated with strong <strong>linkage disequilibrium</strong> using prior expert knowledge, see <a href="http://dougspeed.com/wp-content/uploads/highld.txt">here</a> for details.<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a></p></li>
</ul>
<!-- --><ul>
<li>
<strong>Bash command</strong>:</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="ex">plink2</span> <span class="dt">\</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--bfile</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/merge_5_datasets.eur_only.geno095.hwe1e06.pruned_snps_bfile <span class="dt">\</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--exclude</span> range /mnt/projects_tn01/Cartagene/analyses/QC/high_ld_regions.plink_format.txt <span class="dt">\</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--make-bed</span> <span class="dt">\</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">--out</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/merge_5_datasets.eur_only.geno095.hwe1e06.pruned_snps.high_LD_excluded</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section></section><section id="iv-exclude-affiliated-individuals-based-on-high-ibd-scores" class="level5" data-number="1.3.2.2.4"><h5 data-number="1.3.2.2.4" class="anchored" data-anchor-id="iv-exclude-affiliated-individuals-based-on-high-ibd-scores">
<span class="header-section-number">1.3.2.2.4</span> iv) Exclude affiliated individuals based on high IBD scores</h5>
<ul>
<li>
<p><strong>Inputs</strong>:</p>
<ul>
<li>
<strong>BIM/BED/FAM/hh (Homozygous-Haplotype)</strong> files <code>merge_5_datasets.eur_only.geno095.hwe1e06.pruned_snps_bfile</code> generated from previous local LD trimming, see <a href="#sec-local-LD" class="quarto-xref"><span>Section 1.3.2.2.3.1</span></a>.</li>
</ul>
</li>
<li><p><strong>Objective</strong>: Exclude related individuals computing <strong>identity by descent</strong> (IBD), a genetic metric of the relatedness between two individuals, and exclude one individual by pairs of individuals with a score above <span class="math inline">\(0.2\)</span>. Indeed, strongly associated patients increase bias in GWAS, raising a stronger score than expected in the general population, and prevents quality controls to detect duplicates or sample mix-ups.</p></li>
<li><p><strong>Details</strong>: A two-step, more stringent IBD filtering strategy has been chosen, eliminating <em>first</em> the most problematic individuals, and the <em>second</em> ensuring the remaining related individuals are properly filtered (for each correlated pair, prune randomly one of them). The Bash instructions are reported <a href="./mnt/projects_tn01/Cartagene/analyses/QC/eur_only/related_ids/Readme.txt">here</a>.</p></li>
<li><p><span data-fg="red" style="color: red;"><strong>Remarks</strong>: Choice of a heuristic threshold of <span class="math inline">\(n=68\)</span> affiliated patients removal to be further discussed. Current score of <code>PI_HAT</code> score of <span class="math inline">\(0.2\)</span> is surprising, as common thresholds are either <span class="math inline">\(0.25\)</span> for discarding grandparent-grandchild, or <span class="math inline">\(0.125\)</span> or lower for only keeping the most distant relatives. Starting from <code>PLINK 2.0</code>, the recommended approach is now utilizing the <code>--king-cutoff</code> command, over the older <code>--rel-cutoff</code> and <code>--genome --min</code> + aggregates all subsequent steps simultaneously, especially in heterogeneous populations</span><a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a>.</p></li>
<li><p><strong>Bash command</strong>:</p></li>
</ul>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co">## Calculate IBD, select individuals with IBD above 0.2 for further pruning</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="ex">plink2</span> <span class="dt">\</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--bfile</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/merge_5_datasets.eur_only.geno095.hwe1e06.pruned_snps_bfile <span class="dt">\</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">--genome</span> <span class="dt">\</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">--memory</span> 12006 <span class="dt">\</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">--min</span> 0.2 <span class="dt">\</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">--out</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/merge_5_datasets.eur_only.geno095.hwe1e06.pruned_snps.IBD</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="co">## Remove related individuals, in a two-stage process</span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a><span class="ex">plink2</span> <span class="dt">\</span></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">--bfile</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/merge_5_datasets.eur_only.geno095.hwe1e06.pruned_snps_bfile <span class="dt">\</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">--make-bed</span> <span class="dt">\</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">--out</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/merge_5_datasets.eur_only.geno095.hwe1e06.pruned_snps_bfile.rem_rel_ind_round_1_of_2 <span class="dt">\</span></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">--remove</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/related_ids/IBD.genome.iids.merged.sorted.count.reverse.ids_related_to_2_individuals.ids_only.FID_IID_format</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a><span class="ex">plink2</span> <span class="dt">\</span></span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">--bfile</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/merge_5_datasets.eur_only.geno095.hwe1e06.pruned_snps_bfile.rem_rel_ind_round_1_of_2 <span class="dt">\</span></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">--make-bed</span> <span class="dt">\</span></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">--out</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/merge_5_datasets.eur_only.geno095.hwe1e06.pruned_snps_bfile.rem_rel_ind_round_2_of_2 <span class="dt">\</span></span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">--remove</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/related_ids/IBD.genome2.related_individuals_to_remove.txt </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section></section><section id="sec-PCA" class="level4" data-number="1.3.2.3"><h4 data-number="1.3.2.3" class="anchored" data-anchor-id="sec-PCA">
<span class="header-section-number">1.3.2.3</span> Step 3: PCA</h4>
<ul>
<li>
<p><strong>Inputs</strong>:</p>
<ul>
<li>
<strong>BIM/BED/FAM/hh (Homozygous-Haplotype)</strong> files <code>merge_5_datasets.eur_only.geno095.hwe1e06.pruned_snps.high_LD_excluded</code> resulting from the pre-processing operations reported in <a href="#sec-preprocessing-redundancy" class="quarto-xref"><span>Section 1.3.2.2</span></a> (LD, HWD and missing SNPs trimming)</li>
<li>
<strong>phenotype IDs</strong> listed in <code>merge_5_datasets.eur_only.geno095.hwe1e06.pruned_snps_bfile.rem_rel_ind_round_2_of_2</code>, where all affiliated patients have been removed.</li>
</ul>
</li>
<li><p><strong>Objective</strong>: Compute PCA and keep the 10 first principal components.</p></li>
<li><p><span data-fg="red" style="color: red;"><strong>Remark</strong>: use visualizations, such as <em>scree plots</em>, elbow point or/and <em>Tracy-Widom Test</em> to select the final number of PCs, instead of hard threshold, and on the other hand, scatter plots to identify latent structures. See details <a href="https://www.linkedin.com/posts/joachim-schork_statistics-research-visualanalytics-activity-7251577056570785795-b6Rk">here</a> and <a href="https://www.linkedin.com/feed/update/urn:li:activity:730919293617122508">there</a></span>.</p></li>
<li><p><strong>Bash command</strong>:</p></li>
</ul>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="ex">plink2</span> <span class="dt">\</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--bfile</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/merge_5_datasets.eur_only.geno095.hwe1e06.pruned_snps.high_LD_excluded <span class="dt">\ </span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--keep</span> merge_5_datasets.eur_only.geno095.hwe1e06.pruned_snps_bfile.rem_rel_ind_round_2_of_2.fam <span class="dt">\</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--memory</span> 12006 <span class="dt">\</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">--out</span> /mnt/projects_tn01/Cartagene/analyses/QC/eur_only/merge_5_datasets.eur_only.geno095.hwe1e06.pruned_snps.high_LD_excluded.unrelated_ind.PCA <span class="dt">\</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">--pca</span> 10</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section></section><section id="run-gwas-analyses" class="level3" data-number="1.3.3"><h3 data-number="1.3.3" class="anchored" data-anchor-id="run-gwas-analyses">
<span class="header-section-number">1.3.3</span> Run GWAS analyses</h3>
<section id="sec-phenotype-extraction" class="level4" data-number="1.3.3.1"><h4 data-number="1.3.3.1" class="anchored" data-anchor-id="sec-phenotype-extraction">
<span class="header-section-number">1.3.3.1</span> Step 1: Phenotype feature extraction</h4>
<p>Original phenotype annotations are available in folder <code>/mnt/projects_tn01/Cartagene/analyses/phenotypes</code>. Variables of interest can e downloaded <a href="./data/phenotypes/cartagene_response_variables.xlsx">here</a>:</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cartagene_response_variables</span> <span class="op">&lt;-</span> <span class="fu">readxl</span><span class="fu">::</span><span class="fu"><a href="https://readxl.tidyverse.org/reference/read_excel.html">read_excel</a></span><span class="op">(</span><span class="st">"data/phenotypes/cartagene_response_variables.xlsx"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://davidgohel.github.io/flextable/reference/flextable.html">flextable</a></span><span class="op">(</span><span class="va">cartagene_response_variables</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/bold.html">bold</a></span><span class="op">(</span>part<span class="op">=</span><span class="st">"header"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div id="tbl-cartage-pheno-features" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-layout-align="center">
<figure class="quarto-float quarto-float-tbl figure"><div aria-describedby="tbl-cartage-pheno-features-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<div class="tabwid">
<style>.cl-cbb4de4a{}.cl-cb9dddd0{font-family:'DejaVu Sans';font-size:11pt;font-weight:bold;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-cb9dde02{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-cba7839e{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-cba7eb7c{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-cba7eb9a{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-cba7eba4{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style>
<table data-quarto-disable-processing="true" class="cl-cbb4de4a">
<thead><tr style="overflow-wrap:break-word;">
<th class="cl-cba7eb7c"><p class="cl-cba7839e"><span class="cl-cb9dddd0">Disease</span></p></th>
<th class="cl-cba7eb7c"><p class="cl-cba7839e"><span class="cl-cb9dddd0">Features</span></p></th>
<th class="cl-cba7eb7c"><p class="cl-cba7839e"><span class="cl-cb9dddd0">Type</span></p></th>
<th class="cl-cba7eb7c"><p class="cl-cba7839e"><span class="cl-cb9dddd0">Description</span></p></th>
</tr></thead>
<tbody>
<tr style="overflow-wrap:break-word;">
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Bone Damage</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">OSTEOPONIA</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Boolean</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Exclude osteosecondaire</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Bone Damage</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">OSTEOPOROSIS</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Boolean</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Exclude osteosecondaire</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Bone Damage</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">DMOTSCORE_mod</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Continuous</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Exclude osteosecondaire</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Bone Damage</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">FXALL</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Boolean</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">All fracture events</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Bone Damage</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">FXMOF_POST</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Boolean</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Restriction to osteoporisis events</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Cardiovascular events</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">CVALL_POST</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Boolean</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Post General Cardio event</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Cardiovascular events</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">CVMACE_POST</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Boolean</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Post General Critical Cardio event</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Cardiovascular events</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">MCASGLOBAL_PRE_M</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Boolean</span></p></td>
<td class="cl-cba7eb9a"><p class="cl-cba7839e"><span class="cl-cb9dde02">Pre General Cardio event</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-cba7eba4"><p class="cl-cba7839e"><span class="cl-cb9dde02">Cardiovascular events</span></p></td>
<td class="cl-cba7eba4"><p class="cl-cba7839e"><span class="cl-cb9dde02">AVCGLOBAL_PRE_M</span></p></td>
<td class="cl-cba7eba4"><p class="cl-cba7839e"><span class="cl-cb9dde02">Boolean</span></p></td>
<td class="cl-cba7eba4"><p class="cl-cba7839e"><span class="cl-cb9dde02">Pre General Critical Cardio event</span></p></td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-tbl" id="tbl-cartage-pheno-features-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1.1: Features of interest, and response variables, to predict in <code>CarTaGene</code>.
</figcaption></figure>
</div>
</div>
<section id="i-merge-genotype-ids-phenotypes-and-pcas-values" class="level5" data-number="1.3.3.1.1"><h5 data-number="1.3.3.1.1" class="anchored" data-anchor-id="i-merge-genotype-ids-phenotypes-and-pcas-values">
<span class="header-section-number">1.3.3.1.1</span> i) Merge genotype IDs, phenotypes and PCAs values</h5>
<p>General phenotype features are provided in</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cartagene_phenodata</span> <span class="op">&lt;-</span> <span class="fu">haven</span><span class="fu">::</span><span class="fu"><a href="https://haven.tidyverse.org/reference/read_sas.html">read_sas</a></span><span class="op">(</span><span class="st">"data/phenotypes/cart_mars2025.sas7bdat"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://davidgohel.github.io/flextable/reference/flextable.html">flextable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">cartagene_phenodata</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/bold.html">bold</a></span><span class="op">(</span>bold <span class="op">=</span> <span class="cn">TRUE</span>, part <span class="op">=</span> <span class="st">"header"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div id="tbl-cartage-pheno" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-layout-align="center">
<figure class="quarto-float quarto-float-tbl figure"><div aria-describedby="tbl-cartage-pheno-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<div class="tabwid">
<style>.cl-ccf5a10e{}.cl-ccd324d0{font-family:'DejaVu Sans';font-size:11pt;font-weight:bold;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-ccd324f8{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-cce6dbe2{margin:0;text-align:right;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-cce6dc14{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-cce731dc{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-cce731fa{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-cce7320e{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-cce73218{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-cce73222{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-cce7322c{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style>
<table data-quarto-disable-processing="true" class="cl-ccf5a10e">
<thead><tr style="overflow-wrap:break-word;">
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">PROJECT_CODE</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">ASA</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">STATINE</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">Age (decimal)</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">CKD_EPI</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">Body Mass Index (BMI)</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">T-Score</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">BPS</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">SRAAI</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">GENDER01</span></p></th>
<th class="cl-cce731fa"><p class="cl-cce6dc14"><span class="cl-ccd324d0">ETHNICITY6M</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">TABACCOURANT_M</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">ROHSEMAINE</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">DIABETEGLOBAL</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">DLPGLOBAL</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">HTAGLOBAL</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">ASTHME_MPOCAUTO_RX</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">CIRRHOSEAUTO_M</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">OSTEOAUTO_M</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">OSTEORX</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">NEOALL_AUTO_M</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">NEOCHIMIO_AUTO_M</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">NEORADIO_AUTO_M</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">OSTEOSECONDAIRE</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">CVMACE_POST</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">CVMACE_PROCEDURE_POST</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">CVALL_POST</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">CVALL_POSTDATE</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">CVMACE_POSTDATE</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">CVMACE_PROCEDURE_POSTDATE</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">DECES</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">DATEDECES1216</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">FXALL_POST</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">FXALL_PRE1</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">FXALL_PRE5</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">FXMOF_POST</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">FXALL_POSTDATE</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">FXMOF_POSTDATE</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">AVCGLOBAL_PRE_M</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">ATCDFAMCV</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">HRT</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">IRC</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">PWV1</span></p></th>
<th class="cl-cce731dc"><p class="cl-cce6dbe2"><span class="cl-ccd324d0">MCASGLOBAL_PRE_M</span></p></th>
</tr></thead>
<tbody>
<tr style="overflow-wrap:break-word;">
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">78,295,800,001</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">62.08</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">56.60608</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8"></span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1.7</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73218"><p class="cl-cce6dc14"><span class="cl-ccd324f8">Blanc</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0.25</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,228</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,228</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,228</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8"></span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,228</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,228</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">10.293434</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">78,295,800,002</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">43.84</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">77.58291</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">35.4</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">3.2</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce73218"><p class="cl-cce6dc14"><span class="cl-ccd324f8">Latin</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0.00</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">236</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,059</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">236</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8"></span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,059</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,059</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">6.110750</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">78,295,800,003</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">59.27</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">86.25898</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">22.7</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">-2.3</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce73218"><p class="cl-cce6dc14"><span class="cl-ccd324f8">Blanc</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">8.00</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,205</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,205</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,205</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8"></span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,205</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,205</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">8.271314</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">78,295,800,004</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">58.52</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">85.17528</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">23.5</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">-1.8</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce73218"><p class="cl-cce6dc14"><span class="cl-ccd324f8">Blanc</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0.00</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,184</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,184</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,184</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8"></span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,184</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,184</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">8.810156</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">78,295,800,005</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">54.92</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">105.51846</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">30.3</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1.1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73218"><p class="cl-cce6dc14"><span class="cl-ccd324f8">Blanc</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">4.50</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,103</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,103</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,103</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8"></span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,103</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,103</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">8.310378</span></p></td>
<td class="cl-cce7320e"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">78,295,800,006</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">45.06</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">113.30294</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">28.2</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">-1.2</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce7322c"><p class="cl-cce6dc14"><span class="cl-ccd324f8">Arabe</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0.00</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">1</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,000</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,000</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,000</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8"></span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,000</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">2,000</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8"></span></p></td>
<td class="cl-cce73222"><p class="cl-cce6dbe2"><span class="cl-ccd324f8">0</span></p></td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-tbl" id="tbl-cartage-pheno-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1.2: Read SAS table describing CarTaGene phenotypes.
</figcaption></figure>
</div>
</div>
<p>Number of patients overall is 19990, and number of phenotype variables is 44.</p>
<p>We then need to map each individual patient ID (<code>IID</code>) with its corresponding genotype array <a href="#tbl-cartage-genotypes-IDs" class="quarto-xref">Table&nbsp;<span>1.3</span></a>, as done in <a href="#lst-join-phenotype-genotype" class="quarto-xref">Listing&nbsp;<span>1.1</span></a>.</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cartagene_genotypes_ID</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_csv2</a></span><span class="op">(</span><span class="st">"./data/phenotypes/cartagene_genotype_IDs.csv"</span>,</span>
<span>                                           show_col_types <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                                           col_types <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"d"</span>, <span class="st">"c"</span>,<span class="st">"c"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>PROJECT_CODE <span class="op">=</span> <span class="st">"project_code"</span>, geno_id <span class="op">=</span> <span class="st">"file_111"</span>, batch<span class="op">=</span><span class="st">"batch"</span><span class="op">)</span></span>
<span>  </span>
<span></span>
<span><span class="fu"><a href="https://davidgohel.github.io/flextable/reference/flextable.html">flextable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">cartagene_genotypes_ID</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/bold.html">bold</a></span><span class="op">(</span>bold <span class="op">=</span> <span class="cn">TRUE</span>, part <span class="op">=</span> <span class="st">"header"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div id="tbl-cartage-genotypes-IDs" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-layout-align="center">
<figure class="quarto-float quarto-float-tbl figure"><div aria-describedby="tbl-cartage-genotypes-IDs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<div class="tabwid">
<style>.cl-cd8aac68{}.cl-cd74d23a{font-family:'DejaVu Sans';font-size:11pt;font-weight:bold;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-cd74d262{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-cd7d34f2{margin:0;text-align:right;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-cd7d351a{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-cd7de7ee{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-cd7de80c{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-cd7de82a{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-cd7de83e{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-cd7de848{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-cd7de852{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style>
<table data-quarto-disable-processing="true" class="cl-cd8aac68">
<thead><tr style="overflow-wrap:break-word;">
<th class="cl-cd7de7ee"><p class="cl-cd7d34f2"><span class="cl-cd74d23a">PROJECT_CODE</span></p></th>
<th class="cl-cd7de7ee"><p class="cl-cd7d34f2"><span class="cl-cd74d23a">geno_id</span></p></th>
<th class="cl-cd7de80c"><p class="cl-cd7d351a"><span class="cl-cd74d23a">batch</span></p></th>
</tr></thead>
<tbody>
<tr style="overflow-wrap:break-word;">
<td class="cl-cd7de82a"><p class="cl-cd7d34f2"><span class="cl-cd74d262">78,295,800,001</span></p></td>
<td class="cl-cd7de82a"><p class="cl-cd7d34f2"><span class="cl-cd74d262">11,118,538</span></p></td>
<td class="cl-cd7de83e"><p class="cl-cd7d351a"><span class="cl-cd74d262">gsa.17k</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-cd7de82a"><p class="cl-cd7d34f2"><span class="cl-cd74d262">78,295,800,002</span></p></td>
<td class="cl-cd7de82a"><p class="cl-cd7d34f2"><span class="cl-cd74d262">11,135,721</span></p></td>
<td class="cl-cd7de83e"><p class="cl-cd7d351a"><span class="cl-cd74d262">gsa.17k</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-cd7de82a"><p class="cl-cd7d34f2"><span class="cl-cd74d262">78,295,800,003</span></p></td>
<td class="cl-cd7de82a"><p class="cl-cd7d34f2"><span class="cl-cd74d262">11,129,497</span></p></td>
<td class="cl-cd7de83e"><p class="cl-cd7d351a"><span class="cl-cd74d262">gsa.archi.withIBS</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-cd7de82a"><p class="cl-cd7d34f2"><span class="cl-cd74d262">78,295,800,004</span></p></td>
<td class="cl-cd7de82a"><p class="cl-cd7d34f2"><span class="cl-cd74d262">11,107,817</span></p></td>
<td class="cl-cd7de83e"><p class="cl-cd7d351a"><span class="cl-cd74d262">gsa.17k</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-cd7de82a"><p class="cl-cd7d34f2"><span class="cl-cd74d262">78,295,800,005</span></p></td>
<td class="cl-cd7de82a"><p class="cl-cd7d34f2"><span class="cl-cd74d262">11,131,634</span></p></td>
<td class="cl-cd7de83e"><p class="cl-cd7d351a"><span class="cl-cd74d262">gsa.archi.withIBS</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-cd7de848"><p class="cl-cd7d34f2"><span class="cl-cd74d262">78,295,800,006</span></p></td>
<td class="cl-cd7de848"><p class="cl-cd7d34f2"><span class="cl-cd74d262">11,126,455</span></p></td>
<td class="cl-cd7de852"><p class="cl-cd7d351a"><span class="cl-cd74d262">gsa.17k</span></p></td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-tbl" id="tbl-cartage-genotypes-IDs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1.3: Read SAS table describing CarTaGene phenotypes.
</figcaption></figure>
</div>
</div>
<div class="cell" data-layout-align="center">
<div id="lst-join-phenotype-genotype" class="r cell-code listing quarto-float quarto-figure quarto-figure-left anchored">
<figure class="quarto-float quarto-float-lst figure"><figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-join-phenotype-genotype-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;1.1: Inner join between phenotypes IDs and genotypes, while constraing the remaining individuals to belong to white ethnicity.
</figcaption><div aria-describedby="lst-join-phenotype-genotype-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode" id="lst-join-phenotype-genotype"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cartagene_phenodata</span> <span class="op">&lt;-</span> <span class="va">cartagene_phenodata</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">inner_join</a></span><span class="op">(</span><span class="va">cartagene_genotypes_ID</span>, by<span class="op">=</span><span class="st">"PROJECT_CODE"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">ETHNICITY6M</span><span class="op">==</span><span class="st">"Blanc"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
<p>The resulting phenotype table, after joining with genotypes IDs and restraining to Eurasian phenotypes, stores 17147 individuals.</p>
<p>PCA vectors computed in <a href="#sec-PCA" class="quarto-xref"><span>Section 1.3.2.3</span></a> are subsequently merged with phenotype data in <a href="#lst-join-phenotype-PCA" class="quarto-xref">Listing&nbsp;<span>1.2</span></a>.</p>
<div class="cell" data-layout-align="center">
<div id="lst-join-phenotype-PCA" class="r cell-code listing quarto-float quarto-figure quarto-figure-left anchored">
<figure class="quarto-float quarto-float-lst figure"><figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-join-phenotype-PCA-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;1.2: Inner join between phenotypes IDs and first 10 PCA eigen vectors.
</figcaption><div aria-describedby="lst-join-phenotype-PCA-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode" id="lst-join-phenotype-PCA"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">PCs</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_delim</a></span><span class="op">(</span><span class="st">"data/PCAs/PCA_eigenvec"</span>,</span>
<span>                        col_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"FID"</span>, <span class="st">"geno_id"</span>, <span class="st">"PC1"</span>, <span class="st">"PC2"</span>, <span class="st">"PC3"</span>, <span class="st">"PC4"</span>, <span class="st">"PC5"</span>, <span class="st">"PC6"</span>, <span class="st">"PC7"</span>, <span class="st">"PC8"</span>, <span class="st">"PC9"</span>, <span class="st">"PC10"</span><span class="op">)</span>, </span>
<span>                        delim <span class="op">=</span> <span class="st">" "</span>,</span>
<span>                        show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">cartagene_phenodata</span> <span class="op">&lt;-</span> <span class="va">PCs</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">inner_join</a></span><span class="op">(</span><span class="va">cartagene_phenodata</span>, by<span class="op">=</span><span class="st">"geno_id"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>IID <span class="op">=</span> <span class="va">geno_id</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
<ul>
<li><span data-fg="red" style="color: red;"><strong>Remark</strong>: Avoid using French CSV settings, switch to universal convention, where delimiter is a comma: <code>,</code></span></li>
</ul></section><section id="ii-bone-damage-osteoporosis-and-fractures" class="level5" data-number="1.3.3.1.2"><h5 data-number="1.3.3.1.2" class="anchored" data-anchor-id="ii-bone-damage-osteoporosis-and-fractures">
<span class="header-section-number">1.3.3.1.2</span> ii) Bone damage: osteoporosis and fractures</h5>
<ul>
<li>
<p>The response variables <code>OSTEOPONIA</code> and <code>OSTEOPOROSIS</code> have been computed following these rules <a href="#lst-OSTEOPONIA-computation" class="quarto-xref">Listing&nbsp;<span>1.3</span></a>, with the resulting contingency tables reported in <a href="#tbl-OSTEOPONIA-computation" class="quarto-xref">Table&nbsp;<span>1.4</span></a>.</p>
<ol type="1">
<li>Exclude patients with <code>osteosecondaire==1</code>.</li>
<li>
<code>OSTEOPONIA</code> is <code>case: 1</code> if <code>DMOTSCORE_mod &lt;= -1.5</code> and <code>control</code> elsewhere<a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a>.</li>
<li>
<code>OSTEOPOROSIS</code> is <code>case: 1</code> if <code>DMOTSCORE_mod &lt; -2.5</code> and <code>control</code> elsewhere. <span data-fg="red" style="color: red;"><code>DMOTSCORE_mod</code> has been likely Gaussian standardised + lacks of evidence supporting clear multi-modal distributions. Is there a reporting of the pre-processing operations?</span>.</li>
</ol>
</li>
</ul>
<div class="cell" data-layout-align="center">
<div id="lst-OSTEOPONIA-computation" class="r cell-code listing quarto-float quarto-figure quarto-figure-left anchored">
<figure class="quarto-float quarto-float-lst figure"><figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-OSTEOPONIA-computation-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;1.3: Generate scores of interest.
</figcaption><div aria-describedby="lst-OSTEOPONIA-computation-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode" id="lst-OSTEOPONIA-computation"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cartagene_phenodata</span> <span class="op">&lt;-</span> <span class="va">cartagene_phenodata</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>DMOTSCORE_mod <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/if_else.html">if_else</a></span><span class="op">(</span><span class="va">OSTEOSECONDAIRE</span> <span class="op">==</span><span class="fl">1</span>, <span class="cn">NA</span>, <span class="va">DMOTSCORE</span><span class="op">)</span>, </span>
<span>                OSTEOPONIA <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html">case_when</a></span><span class="op">(</span><span class="va">DMOTSCORE_mod</span> <span class="op">&lt;=</span> <span class="op">-</span><span class="fl">1.5</span> <span class="op">~</span> <span class="fl">1</span>, </span>
<span>                                              <span class="va">DMOTSCORE_mod</span> <span class="op">&gt;</span> <span class="op">-</span><span class="fl">1.5</span> <span class="op">~</span> <span class="fl">0</span>, </span>
<span>                                              .default <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>,</span>
<span>                OSTEOPOROSIS <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html">case_when</a></span><span class="op">(</span><span class="va">DMOTSCORE_mod</span> <span class="op">&lt;</span> <span class="op">-</span><span class="fl">2.5</span> <span class="op">~</span> <span class="fl">1</span>, </span>
<span>                                              <span class="va">DMOTSCORE_mod</span> <span class="op">&gt;=</span> <span class="op">-</span><span class="fl">2.5</span> <span class="op">~</span> <span class="fl">0</span>, </span>
<span>                                              .default <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span><span class="op">)</span> </span>
<span>  </span>
<span></span>
<span><span class="fu">flextable</span><span class="fu">::</span><span class="fu"><a href="https://davidgohel.github.io/flextable/reference/proc_freq.html">proc_freq</a></span><span class="op">(</span><span class="va">cartagene_phenodata</span>, </span>
<span>                     <span class="st">"OSTEOPONIA"</span>, <span class="st">"OSTEOPOROSIS"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
<div id="tbl-OSTEOPONIA-computation" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-layout-align="center">
<figure class="quarto-float quarto-float-tbl figure"><div aria-describedby="tbl-OSTEOPONIA-computation-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<div class="tabwid">
<style>.cl-d00b221a{}.cl-cff890a0{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-cff890be{font-family:'DejaVu Sans';font-size:6.6pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;position: relative;bottom:3.3pt;}.cl-d000214e{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d0002162{margin:0;text-align:center;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d000216c{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d0002180{margin:0;text-align:center;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d000218a{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d0007590{width:1.289in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00075a4{width:1.044in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00075b8{width:1.502in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00075c2{width:1.404in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00075cc{width:1.599in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00075d6{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00075e0{width:1.289in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00075e1{width:1.044in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00075ea{width:1.502in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00075eb{width:1.404in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00075f4{width:1.599in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00075f5{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00075fe{width:1.289in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0007608{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0007612{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0007626{width:1.404in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d000763a{width:1.599in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0007644{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0007645{width:1.289in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0007658{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d000766c{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0007680{width:1.404in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0007694{width:1.599in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d000769e{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00076a8{width:1.289in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00076b2{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00076bc{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00076d0{width:1.404in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00076ee{width:1.599in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00076f8{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0007702{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d000770c{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0007716{width:1.404in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0007720{width:1.599in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0007721{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d000772a{width:1.289in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d000773e{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d000773f{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0007748{width:1.404in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d000775c{width:1.599in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d000775d{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0007766{width:1.289in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0007770{width:1.044in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0007784{width:1.502in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d000778e{width:1.404in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0007798{width:1.599in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d00077a2{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style>
<table data-quarto-disable-processing="true" class="cl-d00b221a">
<thead>
<tr style="overflow-wrap:break-word;">
<th rowspan="2" class="cl-d0007590"><p class="cl-d000214e"><span class="cl-cff890a0">OSTEOPONIA</span></p></th>
<th rowspan="2" class="cl-d00075a4"><p class="cl-d000214e"><span class="cl-cff890a0"></span></p></th>
<th colspan="4" class="cl-d00075b8"><p class="cl-d0002162"><span class="cl-cff890a0">OSTEOPOROSIS</span></p></th>
</tr>
<tr style="overflow-wrap:break-word;">
<th class="cl-d00075ea"><p class="cl-d0002162"><span class="cl-cff890a0">0</span></p></th>
<th class="cl-d00075eb"><p class="cl-d0002162"><span class="cl-cff890a0">1</span></p></th>
<th class="cl-d00075f4"><p class="cl-d0002162"><span class="cl-cff890a0">Missing</span></p></th>
<th class="cl-d00075f5"><p class="cl-d0002162"><span class="cl-cff890a0">Total</span></p></th>
</tr>
</thead>
<tbody>
<tr style="overflow-wrap:break-word;">
<td rowspan="2" class="cl-d00075fe"><p class="cl-d000216c"><span class="cl-cff890a0">0</span></p></td>
<td class="cl-d0007608"><p class="cl-d000216c"><span class="cl-cff890a0">Count</span></p></td>
<td class="cl-d0007612"><p class="cl-d0002180"><span class="cl-cff890a0">12,463 (74.8%)</span><span class="cl-cff890a0"></span></p></td>
<td class="cl-d0007626"><p class="cl-d0002180"><span class="cl-cff890a0"></span><span class="cl-cff890a0"></span></p></td>
<td class="cl-d000763a"><p class="cl-d0002180"><span class="cl-cff890a0"></span><span class="cl-cff890a0"></span></p></td>
<td class="cl-d0007644"><p class="cl-d0002180"><span class="cl-cff890a0">12,463 (74.8%)</span><span class="cl-cff890a0"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-d0007658"><p class="cl-d000216c"><span class="cl-cff890a0">Mar. pct</span><span class="cl-cff890be"> (1)</span></p></td>
<td class="cl-d000766c"><p class="cl-d0002180"><span class="cl-cff890a0"></span><span class="cl-cff890a0">94.3% ; 100.0%</span></p></td>
<td class="cl-d0007680"><p class="cl-d0002180"><span class="cl-cff890a0"></span><span class="cl-cff890a0"></span></p></td>
<td class="cl-d0007694"><p class="cl-d0002180"><span class="cl-cff890a0"></span><span class="cl-cff890a0"></span></p></td>
<td class="cl-d000769e"><p class="cl-d0002180"><span class="cl-cff890a0"></span><span class="cl-cff890a0"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td rowspan="2" class="cl-d00075fe"><p class="cl-d000216c"><span class="cl-cff890a0">1</span></p></td>
<td class="cl-d0007608"><p class="cl-d000216c"><span class="cl-cff890a0">Count</span></p></td>
<td class="cl-d0007612"><p class="cl-d0002180"><span class="cl-cff890a0">753 (4.5%)</span><span class="cl-cff890a0"></span></p></td>
<td class="cl-d0007626"><p class="cl-d0002180"><span class="cl-cff890a0">48 (0.3%)</span><span class="cl-cff890a0"></span></p></td>
<td class="cl-d000763a"><p class="cl-d0002180"><span class="cl-cff890a0"></span><span class="cl-cff890a0"></span></p></td>
<td class="cl-d0007644"><p class="cl-d0002180"><span class="cl-cff890a0">801 (4.8%)</span><span class="cl-cff890a0"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-d0007658"><p class="cl-d000216c"><span class="cl-cff890a0">Mar. pct</span></p></td>
<td class="cl-d000766c"><p class="cl-d0002180"><span class="cl-cff890a0"></span><span class="cl-cff890a0">5.7% ; 94.0%</span></p></td>
<td class="cl-d0007680"><p class="cl-d0002180"><span class="cl-cff890a0"></span><span class="cl-cff890a0">100.0% ; 6.0%</span></p></td>
<td class="cl-d0007694"><p class="cl-d0002180"><span class="cl-cff890a0"></span><span class="cl-cff890a0"></span></p></td>
<td class="cl-d000769e"><p class="cl-d0002180"><span class="cl-cff890a0"></span><span class="cl-cff890a0"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td rowspan="2" class="cl-d00076a8"><p class="cl-d000216c"><span class="cl-cff890a0">Missing</span></p></td>
<td class="cl-d00076b2"><p class="cl-d000216c"><span class="cl-cff890a0">Count</span></p></td>
<td class="cl-d00076bc"><p class="cl-d0002180"><span class="cl-cff890a0"></span><span class="cl-cff890a0"></span></p></td>
<td class="cl-d00076d0"><p class="cl-d0002180"><span class="cl-cff890a0"></span><span class="cl-cff890a0"></span></p></td>
<td class="cl-d00076ee"><p class="cl-d0002180"><span class="cl-cff890a0">3,393 (20.4%)</span><span class="cl-cff890a0"></span></p></td>
<td class="cl-d00076f8"><p class="cl-d0002180"><span class="cl-cff890a0">3,393 (20.4%)</span><span class="cl-cff890a0"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-d0007702"><p class="cl-d000216c"><span class="cl-cff890a0">Mar. pct</span></p></td>
<td class="cl-d000770c"><p class="cl-d0002180"><span class="cl-cff890a0"></span><span class="cl-cff890a0"></span></p></td>
<td class="cl-d0007716"><p class="cl-d0002180"><span class="cl-cff890a0"></span><span class="cl-cff890a0"></span></p></td>
<td class="cl-d0007720"><p class="cl-d0002180"><span class="cl-cff890a0"></span><span class="cl-cff890a0">100.0% ; 100.0%</span></p></td>
<td class="cl-d0007721"><p class="cl-d0002180"><span class="cl-cff890a0"></span><span class="cl-cff890a0"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-d000772a"><p class="cl-d000216c"><span class="cl-cff890a0">Total</span></p></td>
<td class="cl-d000773e"><p class="cl-d000216c"><span class="cl-cff890a0">Count</span></p></td>
<td class="cl-d000773f"><p class="cl-d0002180"><span class="cl-cff890a0">13,216 (79.3%)</span><span class="cl-cff890a0"></span></p></td>
<td class="cl-d0007748"><p class="cl-d0002180"><span class="cl-cff890a0">48 (0.3%)</span><span class="cl-cff890a0"></span></p></td>
<td class="cl-d000775c"><p class="cl-d0002180"><span class="cl-cff890a0">3,393 (20.4%)</span><span class="cl-cff890a0"></span></p></td>
<td class="cl-d000775d"><p class="cl-d0002180"><span class="cl-cff890a0">16,657 (100.0%)</span><span class="cl-cff890a0"></span></p></td>
</tr>
</tbody>
<tfoot><tr style="overflow-wrap:break-word;"><td colspan="6" class="cl-d0007766"><p class="cl-d000218a"><span class="cl-cff890be"> (1)</span><span class="cl-cff890a0"> Columns and rows percentages</span></p></td></tr></tfoot>
</table>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-tbl" id="tbl-OSTEOPONIA-computation-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1.4: OSTEOPOROSIS is considered as more severe than <code>OSTEOPONIA</code>, hence the striclty lower number of individuals affected by the disease.
</figcaption></figure>
</div>
</div>
<ul>
<li>
<p><strong>Fracture events</strong>:</p>
<ol type="1">
<li>We merge <code>FXALL_PRE5</code> (any fracture occurring within the five years prior to recruitment) with <code>FXALL_POST</code> (all fractures occurring during follow-up) into variable <code>FXALL</code>, being positive if an event fracture occured prior or posterior. <span data-fg="red" style="color: red;">Which variable between <code>FXALL_PRE5</code> and <code>FXALL_PRE1</code> should be considered? Variables differ by 471 <span class="math inline">\(\%\)</span> overall!!</span>
</li>
<li>
<code>FXMOF_POST</code> (only osteoporotic fractures during follow-up)<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a> <span data-fg="red" style="color: red;">In contrast with labelling, osteoporotic events, as stored in <code>FXMOF_POST</code>, contain both pre-, and post- fracture events, as shown by <a href="#fig-venn-osteo" class="quarto-xref">Figure&nbsp;<span>1.2</span></a>.</span>
</li>
</ol>
</li>
</ul>
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cartagene_phenodata</span> <span class="op">&lt;-</span> <span class="va">cartagene_phenodata</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>FXALL <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/if_else.html">if_else</a></span><span class="op">(</span><span class="va">FXALL_POST</span> <span class="op">==</span> <span class="fl">1L</span> <span class="op">|</span> <span class="va">FXALL_PRE1</span> <span class="op">==</span><span class="fl">1L</span>, <span class="fl">1L</span>, <span class="fl">0L</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">flextable</span><span class="fu">::</span><span class="fu"><a href="https://davidgohel.github.io/flextable/reference/proc_freq.html">proc_freq</a></span><span class="op">(</span><span class="va">cartagene_phenodata</span>, </span>
<span>                     <span class="st">"FXALL"</span>, <span class="st">"FXMOF_POST"</span><span class="op">)</span></span>
<span><span class="fu">flextable</span><span class="fu">::</span><span class="fu"><a href="https://davidgohel.github.io/flextable/reference/proc_freq.html">proc_freq</a></span><span class="op">(</span><span class="va">cartagene_phenodata</span>, </span>
<span>                     <span class="st">"FXALL_POST"</span>, <span class="st">"FXALL_PRE1"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div id="tbl-fracture-computation" class="quarto-layout-panel anchored">
<figure class="quarto-float quarto-float-tbl figure"><div aria-describedby="tbl-fracture-computation-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="tbl-fracture-computation" style="flex-basis: 50.0%;justify-content: center;">
<div id="tbl-fracture-computation-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-tbl figure"><div aria-describedby="tbl-fracture-computation-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<div class="tabwid">
<style>.cl-d0cf577a{}.cl-d0bbe8d4{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-d0bbe8f2{font-family:'DejaVu Sans';font-size:6.6pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;position: relative;bottom:3.3pt;}.cl-d0c3e638{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d0c3e656{margin:0;text-align:center;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d0c3e66a{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d0c3e674{margin:0;text-align:center;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d0c3e67e{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d0c440ce{width:0.752in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c440ec{width:1.044in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c44100{width:1.502in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c44101{width:1.502in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c4410a{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c4411e{width:0.752in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c44128{width:1.044in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c4413c{width:1.502in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c44146{width:1.502in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c4415a{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c44164{width:0.752in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c44182{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c4418c{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c44196{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c44197{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c441a0{width:0.752in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c441aa{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c441be{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c441bf{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c441c8{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c441d2{width:0.752in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c441dc{width:0.752in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c441dd{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c441e6{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c441fa{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c44204{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c4420e{width:0.752in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c4420f{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c44222{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c4422c{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c44236{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c44240{width:0.752in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c44254{width:1.044in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c44255{width:1.502in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c44290{width:1.502in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d0c4429a{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style>
<table data-quarto-disable-processing="true" class="cl-d0cf577a">
<thead>
<tr style="overflow-wrap:break-word;">
<th rowspan="2" class="cl-d0c440ce"><p class="cl-d0c3e638"><span class="cl-d0bbe8d4">FXALL</span></p></th>
<th rowspan="2" class="cl-d0c440ec"><p class="cl-d0c3e638"><span class="cl-d0bbe8d4"></span></p></th>
<th colspan="3" class="cl-d0c44100"><p class="cl-d0c3e656"><span class="cl-d0bbe8d4">FXMOF_POST</span></p></th>
</tr>
<tr style="overflow-wrap:break-word;">
<th class="cl-d0c4413c"><p class="cl-d0c3e656"><span class="cl-d0bbe8d4">0</span></p></th>
<th class="cl-d0c44146"><p class="cl-d0c3e656"><span class="cl-d0bbe8d4">1</span></p></th>
<th class="cl-d0c4415a"><p class="cl-d0c3e656"><span class="cl-d0bbe8d4">Total</span></p></th>
</tr>
</thead>
<tbody>
<tr style="overflow-wrap:break-word;">
<td rowspan="2" class="cl-d0c44164"><p class="cl-d0c3e66a"><span class="cl-d0bbe8d4">0</span></p></td>
<td class="cl-d0c44182"><p class="cl-d0c3e66a"><span class="cl-d0bbe8d4">Count</span></p></td>
<td class="cl-d0c4418c"><p class="cl-d0c3e674"><span class="cl-d0bbe8d4">15,824 (95.0%)</span><span class="cl-d0bbe8d4"></span></p></td>
<td class="cl-d0c44196"><p class="cl-d0c3e674"><span class="cl-d0bbe8d4"></span><span class="cl-d0bbe8d4"></span></p></td>
<td class="cl-d0c44197"><p class="cl-d0c3e674"><span class="cl-d0bbe8d4">15,824 (95.0%)</span><span class="cl-d0bbe8d4"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-d0c441aa"><p class="cl-d0c3e66a"><span class="cl-d0bbe8d4">Mar. pct</span><span class="cl-d0bbe8f2"> (1)</span></p></td>
<td class="cl-d0c441be"><p class="cl-d0c3e674"><span class="cl-d0bbe8d4"></span><span class="cl-d0bbe8d4">96.8% ; 100.0%</span></p></td>
<td class="cl-d0c441bf"><p class="cl-d0c3e674"><span class="cl-d0bbe8d4"></span><span class="cl-d0bbe8d4"></span></p></td>
<td class="cl-d0c441c8"><p class="cl-d0c3e674"><span class="cl-d0bbe8d4"></span><span class="cl-d0bbe8d4"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td rowspan="2" class="cl-d0c441d2"><p class="cl-d0c3e66a"><span class="cl-d0bbe8d4">1</span></p></td>
<td class="cl-d0c44182"><p class="cl-d0c3e66a"><span class="cl-d0bbe8d4">Count</span></p></td>
<td class="cl-d0c4418c"><p class="cl-d0c3e674"><span class="cl-d0bbe8d4">529 (3.2%)</span><span class="cl-d0bbe8d4"></span></p></td>
<td class="cl-d0c44196"><p class="cl-d0c3e674"><span class="cl-d0bbe8d4">304 (1.8%)</span><span class="cl-d0bbe8d4"></span></p></td>
<td class="cl-d0c44197"><p class="cl-d0c3e674"><span class="cl-d0bbe8d4">833 (5.0%)</span><span class="cl-d0bbe8d4"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-d0c441dd"><p class="cl-d0c3e66a"><span class="cl-d0bbe8d4">Mar. pct</span></p></td>
<td class="cl-d0c441e6"><p class="cl-d0c3e674"><span class="cl-d0bbe8d4"></span><span class="cl-d0bbe8d4">3.2% ; 63.5%</span></p></td>
<td class="cl-d0c441fa"><p class="cl-d0c3e674"><span class="cl-d0bbe8d4"></span><span class="cl-d0bbe8d4">100.0% ; 36.5%</span></p></td>
<td class="cl-d0c44204"><p class="cl-d0c3e674"><span class="cl-d0bbe8d4"></span><span class="cl-d0bbe8d4"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-d0c4420e"><p class="cl-d0c3e66a"><span class="cl-d0bbe8d4">Total</span></p></td>
<td class="cl-d0c4420f"><p class="cl-d0c3e66a"><span class="cl-d0bbe8d4">Count</span></p></td>
<td class="cl-d0c44222"><p class="cl-d0c3e674"><span class="cl-d0bbe8d4">16,353 (98.2%)</span><span class="cl-d0bbe8d4"></span></p></td>
<td class="cl-d0c4422c"><p class="cl-d0c3e674"><span class="cl-d0bbe8d4">304 (1.8%)</span><span class="cl-d0bbe8d4"></span></p></td>
<td class="cl-d0c44236"><p class="cl-d0c3e674"><span class="cl-d0bbe8d4">16,657 (100.0%)</span><span class="cl-d0bbe8d4"></span></p></td>
</tr>
</tbody>
<tfoot><tr style="overflow-wrap:break-word;"><td colspan="5" class="cl-d0c44240"><p class="cl-d0c3e67e"><span class="cl-d0bbe8f2"> (1)</span><span class="cl-d0bbe8d4"> Columns and rows percentages</span></p></td></tr></tfoot>
</table>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-tbl" id="tbl-fracture-computation-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) All fractures vs osteporotic fractures
</figcaption></figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="tbl-fracture-computation" style="flex-basis: 50.0%;justify-content: center;">
<div id="tbl-fracture-computation-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-tbl figure"><div aria-describedby="tbl-fracture-computation-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<div class="tabwid">
<style>.cl-d17341fa{}.cl-d15faeba{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-d15faee2{font-family:'DejaVu Sans';font-size:6.6pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;position: relative;bottom:3.3pt;}.cl-d167c46a{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d167c49c{margin:0;text-align:center;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d167c49d{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d167c4a6{margin:0;text-align:center;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d167c4b0{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d1681a00{width:1.232in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681a0a{width:1.044in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681a14{width:1.473in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681a28{width:1.307in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681a32{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681a3c{width:1.232in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681a46{width:1.044in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681a5a{width:1.473in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681a6e{width:1.307in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681a82{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681a8c{width:1.232in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681a96{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681aa0{width:1.473in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681aaa{width:1.307in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681abe{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681ad2{width:1.232in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681ae6{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681afa{width:1.473in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681b04{width:1.307in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681b0e{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681b22{width:1.232in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681b2c{width:1.232in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681b36{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681b40{width:1.473in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681b41{width:1.307in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681b54{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681b68{width:1.232in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681b72{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681b86{width:1.473in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681be0{width:1.307in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681bea{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681beb{width:1.232in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681bf4{width:1.044in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681bfe{width:1.473in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681bff{width:1.307in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d1681c08{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style>
<table data-quarto-disable-processing="true" class="cl-d17341fa">
<thead>
<tr style="overflow-wrap:break-word;">
<th rowspan="2" class="cl-d1681a00"><p class="cl-d167c46a"><span class="cl-d15faeba">FXALL_POST</span></p></th>
<th rowspan="2" class="cl-d1681a0a"><p class="cl-d167c46a"><span class="cl-d15faeba"></span></p></th>
<th colspan="3" class="cl-d1681a14"><p class="cl-d167c49c"><span class="cl-d15faeba">FXALL_PRE1</span></p></th>
</tr>
<tr style="overflow-wrap:break-word;">
<th class="cl-d1681a5a"><p class="cl-d167c49c"><span class="cl-d15faeba">0</span></p></th>
<th class="cl-d1681a6e"><p class="cl-d167c49c"><span class="cl-d15faeba">1</span></p></th>
<th class="cl-d1681a82"><p class="cl-d167c49c"><span class="cl-d15faeba">Total</span></p></th>
</tr>
</thead>
<tbody>
<tr style="overflow-wrap:break-word;">
<td rowspan="2" class="cl-d1681a8c"><p class="cl-d167c49d"><span class="cl-d15faeba">0</span></p></td>
<td class="cl-d1681a96"><p class="cl-d167c49d"><span class="cl-d15faeba">Count</span></p></td>
<td class="cl-d1681aa0"><p class="cl-d167c4a6"><span class="cl-d15faeba">15,824 (95.0%)</span><span class="cl-d15faeba"></span></p></td>
<td class="cl-d1681aaa"><p class="cl-d167c4a6"><span class="cl-d15faeba">92 (0.6%)</span><span class="cl-d15faeba"></span></p></td>
<td class="cl-d1681abe"><p class="cl-d167c4a6"><span class="cl-d15faeba">15,916 (95.6%)</span><span class="cl-d15faeba"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-d1681ae6"><p class="cl-d167c49d"><span class="cl-d15faeba">Mar. pct</span><span class="cl-d15faee2"> (1)</span></p></td>
<td class="cl-d1681afa"><p class="cl-d167c4a6"><span class="cl-d15faeba"></span><span class="cl-d15faeba">95.6% ; 99.4%</span></p></td>
<td class="cl-d1681b04"><p class="cl-d167c4a6"><span class="cl-d15faeba"></span><span class="cl-d15faeba">82.9% ; 0.6%</span></p></td>
<td class="cl-d1681b0e"><p class="cl-d167c4a6"><span class="cl-d15faeba"></span><span class="cl-d15faeba"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td rowspan="2" class="cl-d1681b22"><p class="cl-d167c49d"><span class="cl-d15faeba">1</span></p></td>
<td class="cl-d1681a96"><p class="cl-d167c49d"><span class="cl-d15faeba">Count</span></p></td>
<td class="cl-d1681aa0"><p class="cl-d167c4a6"><span class="cl-d15faeba">722 (4.3%)</span><span class="cl-d15faeba"></span></p></td>
<td class="cl-d1681aaa"><p class="cl-d167c4a6"><span class="cl-d15faeba">19 (0.1%)</span><span class="cl-d15faeba"></span></p></td>
<td class="cl-d1681abe"><p class="cl-d167c4a6"><span class="cl-d15faeba">741 (4.4%)</span><span class="cl-d15faeba"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-d1681b36"><p class="cl-d167c49d"><span class="cl-d15faeba">Mar. pct</span></p></td>
<td class="cl-d1681b40"><p class="cl-d167c4a6"><span class="cl-d15faeba"></span><span class="cl-d15faeba">4.4% ; 97.4%</span></p></td>
<td class="cl-d1681b41"><p class="cl-d167c4a6"><span class="cl-d15faeba"></span><span class="cl-d15faeba">17.1% ; 2.6%</span></p></td>
<td class="cl-d1681b54"><p class="cl-d167c4a6"><span class="cl-d15faeba"></span><span class="cl-d15faeba"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-d1681b68"><p class="cl-d167c49d"><span class="cl-d15faeba">Total</span></p></td>
<td class="cl-d1681b72"><p class="cl-d167c49d"><span class="cl-d15faeba">Count</span></p></td>
<td class="cl-d1681b86"><p class="cl-d167c4a6"><span class="cl-d15faeba">16,546 (99.3%)</span><span class="cl-d15faeba"></span></p></td>
<td class="cl-d1681be0"><p class="cl-d167c4a6"><span class="cl-d15faeba">111 (0.7%)</span><span class="cl-d15faeba"></span></p></td>
<td class="cl-d1681bea"><p class="cl-d167c4a6"><span class="cl-d15faeba">16,657 (100.0%)</span><span class="cl-d15faeba"></span></p></td>
</tr>
</tbody>
<tfoot><tr style="overflow-wrap:break-word;"><td colspan="5" class="cl-d1681beb"><p class="cl-d167c4b0"><span class="cl-d15faee2"> (1)</span><span class="cl-d15faeba"> Columns and rows percentages</span></p></td></tr></tfoot>
</table>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-tbl" id="tbl-fracture-computation-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Pre vs post-fractures.
</figcaption></figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-tbl" id="tbl-fracture-computation-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1.5: Description of fracture events.
</figcaption></figure>
</div>
<div id="fig-venn-osteo" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-venn-osteo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-venn-osteo" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-venn-osteo-1" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-subfloat-fig figure"><div aria-describedby="fig-venn-osteo-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="CARTaGENE-GWAS-Report_files/figure-html/fig-venn-osteo-1.png" class="lightbox" data-gallery="fig-venn-osteo" title="Figure&nbsp;1.2&nbsp;(a): Generated by VennDiagram"><img src="CARTaGENE-GWAS-Report_files/figure-html/fig-venn-osteo-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" data-ref-parent="fig-venn-osteo" width="672"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-venn-osteo-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) Generated by <code>VennDiagram</code>
</figcaption></figure>
</div>
</div>
<div class="cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-venn-osteo" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-venn-osteo-2" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-subfloat-fig figure"><div aria-describedby="fig-venn-osteo-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="figures/venn_diagramm_fractures.png" class="lightbox" data-gallery="fig-venn-osteo" title="Figure&nbsp;1.2&nbsp;(b): Generated by ggVennDiagram"><img src="figures/venn_diagramm_fractures.png" class="img-fluid quarto-figure quarto-figure-center figure-img" data-ref-parent="fig-venn-osteo" width="240"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-venn-osteo-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Generated by <code>ggVennDiagram</code>
</figcaption></figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-venn-osteo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1.2: Venn Diagrams of fracture events, before and after diagnosis control.
</figcaption></figure>
</div>
</section><section id="iii-cardiovascular-diseases" class="level5" data-number="1.3.3.1.3"><h5 data-number="1.3.3.1.3" class="anchored" data-anchor-id="iii-cardiovascular-diseases">
<span class="header-section-number">1.3.3.1.3</span> iii) Cardiovascular diseases</h5>
<!-- - The variable `PWV1`, newly included in the dataset, represents an algorithmic estimation of pulse wave velocity. It is a continuous measure positively correlated with arterial stiffness. -->
<ul>
<li><p>The variables <code>CVALL_POST</code> and <code>CVMACE_POST</code> correspond to the occurrence of any cardiovascular event and <strong>major adverse cardiovascular events</strong> (MACE), including myocardial infarction, stroke, or cardiovascular death events, is the cognate subset of the most life-threatening heart conditions. <span data-fg="red" style="color: red;">An extended follow-up dataset is available for <code>CVALL_POST</code>; however, it is not currently accessible.</span></p></li>
<li><p>The variable <code>MCASGLOBAL_PRE_M</code> captures ischemic heart disease events, at baseline.</p></li>
<li><p>The variable <code>AVCGLOBAL_PRE_M</code> represents the history of stroke events.</p></li>
</ul>
<p>Contingency tables of both pre-, and post-, cardiovascular events are reported in <a href="#tbl-cardiovascular-diseases" class="quarto-xref">Table&nbsp;<span>1.6</span></a>.</p>
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">flextable</span><span class="fu">::</span><span class="fu"><a href="https://davidgohel.github.io/flextable/reference/proc_freq.html">proc_freq</a></span><span class="op">(</span><span class="va">cartagene_phenodata</span>, </span>
<span>                     <span class="st">"CVALL_POST"</span>, <span class="st">"CVMACE_POST"</span><span class="op">)</span></span>
<span><span class="fu">flextable</span><span class="fu">::</span><span class="fu"><a href="https://davidgohel.github.io/flextable/reference/proc_freq.html">proc_freq</a></span><span class="op">(</span><span class="va">cartagene_phenodata</span>, </span>
<span>                     <span class="st">"MCASGLOBAL_PRE_M"</span>, <span class="st">"AVCGLOBAL_PRE_M"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div id="tbl-cardiovascular-diseases" class="quarto-layout-panel anchored">
<figure class="quarto-float quarto-float-tbl figure"><div aria-describedby="tbl-cardiovascular-diseases-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="tbl-cardiovascular-diseases" style="flex-basis: 50.0%;justify-content: center;">
<div id="tbl-cardiovascular-diseases-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-tbl figure"><div aria-describedby="tbl-cardiovascular-diseases-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<div class="tabwid">
<style>.cl-d35ccec8{}.cl-d348be10{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-d348be4c{font-family:'DejaVu Sans';font-size:6.6pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;position: relative;bottom:3.3pt;}.cl-d3510f02{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d3510f20{margin:0;text-align:center;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d3510f2a{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d3510f3e{margin:0;text-align:center;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d3510f3f{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d3517082{width:1.25in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d35170aa{width:1.044in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d35170be{width:1.502in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d35170d2{width:1.502in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d35170d3{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d35170dc{width:1.25in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d35170f0{width:1.044in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d3517104{width:1.502in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d351710e{width:1.502in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d351710f{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d3517118{width:1.25in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d351712c{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d351712d{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d3517140{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d3517154{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d351715e{width:1.25in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d3517168{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d3517172{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d351717c{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d3517186{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d3517190{width:1.25in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d351719a{width:1.25in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d35171a4{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d35171a5{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d35171ae{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d35171c2{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d35171cc{width:1.25in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d35171e0{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d35171ea{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d35171f4{width:1.502in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d35171fe{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d35171ff{width:1.25in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d3517226{width:1.044in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d351723a{width:1.502in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d351724e{width:1.502in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d3517258{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style>
<table data-quarto-disable-processing="true" class="cl-d35ccec8">
<thead>
<tr style="overflow-wrap:break-word;">
<th rowspan="2" class="cl-d3517082"><p class="cl-d3510f02"><span class="cl-d348be10">CVALL_POST</span></p></th>
<th rowspan="2" class="cl-d35170aa"><p class="cl-d3510f02"><span class="cl-d348be10"></span></p></th>
<th colspan="3" class="cl-d35170be"><p class="cl-d3510f20"><span class="cl-d348be10">CVMACE_POST</span></p></th>
</tr>
<tr style="overflow-wrap:break-word;">
<th class="cl-d3517104"><p class="cl-d3510f20"><span class="cl-d348be10">0</span></p></th>
<th class="cl-d351710e"><p class="cl-d3510f20"><span class="cl-d348be10">1</span></p></th>
<th class="cl-d351710f"><p class="cl-d3510f20"><span class="cl-d348be10">Total</span></p></th>
</tr>
</thead>
<tbody>
<tr style="overflow-wrap:break-word;">
<td rowspan="2" class="cl-d3517118"><p class="cl-d3510f2a"><span class="cl-d348be10">0</span></p></td>
<td class="cl-d351712c"><p class="cl-d3510f2a"><span class="cl-d348be10">Count</span></p></td>
<td class="cl-d351712d"><p class="cl-d3510f3e"><span class="cl-d348be10">16,215 (97.3%)</span><span class="cl-d348be10"></span></p></td>
<td class="cl-d3517140"><p class="cl-d3510f3e"><span class="cl-d348be10"></span><span class="cl-d348be10"></span></p></td>
<td class="cl-d3517154"><p class="cl-d3510f3e"><span class="cl-d348be10">16,215 (97.3%)</span><span class="cl-d348be10"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-d3517168"><p class="cl-d3510f2a"><span class="cl-d348be10">Mar. pct</span><span class="cl-d348be4c"> (1)</span></p></td>
<td class="cl-d3517172"><p class="cl-d3510f3e"><span class="cl-d348be10"></span><span class="cl-d348be10">98.7% ; 100.0%</span></p></td>
<td class="cl-d351717c"><p class="cl-d3510f3e"><span class="cl-d348be10"></span><span class="cl-d348be10"></span></p></td>
<td class="cl-d3517186"><p class="cl-d3510f3e"><span class="cl-d348be10"></span><span class="cl-d348be10"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td rowspan="2" class="cl-d3517190"><p class="cl-d3510f2a"><span class="cl-d348be10">1</span></p></td>
<td class="cl-d351712c"><p class="cl-d3510f2a"><span class="cl-d348be10">Count</span></p></td>
<td class="cl-d351712d"><p class="cl-d3510f3e"><span class="cl-d348be10">221 (1.3%)</span><span class="cl-d348be10"></span></p></td>
<td class="cl-d3517140"><p class="cl-d3510f3e"><span class="cl-d348be10">221 (1.3%)</span><span class="cl-d348be10"></span></p></td>
<td class="cl-d3517154"><p class="cl-d3510f3e"><span class="cl-d348be10">442 (2.7%)</span><span class="cl-d348be10"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-d35171a4"><p class="cl-d3510f2a"><span class="cl-d348be10">Mar. pct</span></p></td>
<td class="cl-d35171a5"><p class="cl-d3510f3e"><span class="cl-d348be10"></span><span class="cl-d348be10">1.3% ; 50.0%</span></p></td>
<td class="cl-d35171ae"><p class="cl-d3510f3e"><span class="cl-d348be10"></span><span class="cl-d348be10">100.0% ; 50.0%</span></p></td>
<td class="cl-d35171c2"><p class="cl-d3510f3e"><span class="cl-d348be10"></span><span class="cl-d348be10"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-d35171cc"><p class="cl-d3510f2a"><span class="cl-d348be10">Total</span></p></td>
<td class="cl-d35171e0"><p class="cl-d3510f2a"><span class="cl-d348be10">Count</span></p></td>
<td class="cl-d35171ea"><p class="cl-d3510f3e"><span class="cl-d348be10">16,436 (98.7%)</span><span class="cl-d348be10"></span></p></td>
<td class="cl-d35171f4"><p class="cl-d3510f3e"><span class="cl-d348be10">221 (1.3%)</span><span class="cl-d348be10"></span></p></td>
<td class="cl-d35171fe"><p class="cl-d3510f3e"><span class="cl-d348be10">16,657 (100.0%)</span><span class="cl-d348be10"></span></p></td>
</tr>
</tbody>
<tfoot><tr style="overflow-wrap:break-word;"><td colspan="5" class="cl-d35171ff"><p class="cl-d3510f3f"><span class="cl-d348be4c"> (1)</span><span class="cl-d348be10"> Columns and rows percentages</span></p></td></tr></tfoot>
</table>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-tbl" id="tbl-cardiovascular-diseases-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) CVALL_POST vs CVMACE_POST
</figcaption></figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="tbl-cardiovascular-diseases" style="flex-basis: 50.0%;justify-content: center;">
<div id="tbl-cardiovascular-diseases-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-tbl figure"><div aria-describedby="tbl-cardiovascular-diseases-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<div class="tabwid">
<style>.cl-d416c47c{}.cl-d40442a2{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-d40442ca{font-family:'DejaVu Sans';font-size:6.6pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;position: relative;bottom:3.3pt;}.cl-d40bd242{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d40bd256{margin:0;text-align:center;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d40bd26a{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d40bd274{margin:0;text-align:center;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d40bd27e{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-d40c2850{width:1.923in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c2864{width:1.044in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c2882{width:1.473in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c288c{width:1.307in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c28a0{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c28aa{width:1.923in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c28b4{width:1.044in;background-color:transparent;vertical-align: bottom;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c28b5{width:1.473in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c28d2{width:1.307in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c28dc{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c28e6{width:1.923in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c28f0{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c2904{width:1.473in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c290e{width:1.307in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c2918{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c2922{width:1.923in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c2923{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c292c{width:1.473in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c2936{width:1.307in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c294a{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c295e{width:1.923in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c295f{width:1.923in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c2972{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c297c{width:1.473in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c2990{width:1.307in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c2991{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c299a{width:1.923in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c29a4{width:1.044in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c29a5{width:1.473in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c29b8{width:1.307in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c29cc{width:1.57in;background-color:transparent;vertical-align: top;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c29d6{width:1.923in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c29e0{width:1.044in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c29ea{width:1.473in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c29eb{width:1.307in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 1pt solid rgba(102, 102, 102, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-d40c29fe{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 1pt solid rgba(102, 102, 102, 1.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style>
<table data-quarto-disable-processing="true" class="cl-d416c47c">
<thead>
<tr style="overflow-wrap:break-word;">
<th rowspan="2" class="cl-d40c2850"><p class="cl-d40bd242"><span class="cl-d40442a2">MCASGLOBAL_PRE_M</span></p></th>
<th rowspan="2" class="cl-d40c2864"><p class="cl-d40bd242"><span class="cl-d40442a2"></span></p></th>
<th colspan="3" class="cl-d40c2882"><p class="cl-d40bd256"><span class="cl-d40442a2">AVCGLOBAL_PRE_M</span></p></th>
</tr>
<tr style="overflow-wrap:break-word;">
<th class="cl-d40c28b5"><p class="cl-d40bd256"><span class="cl-d40442a2">0</span></p></th>
<th class="cl-d40c28d2"><p class="cl-d40bd256"><span class="cl-d40442a2">1</span></p></th>
<th class="cl-d40c28dc"><p class="cl-d40bd256"><span class="cl-d40442a2">Total</span></p></th>
</tr>
</thead>
<tbody>
<tr style="overflow-wrap:break-word;">
<td rowspan="2" class="cl-d40c28e6"><p class="cl-d40bd26a"><span class="cl-d40442a2">0</span></p></td>
<td class="cl-d40c28f0"><p class="cl-d40bd26a"><span class="cl-d40442a2">Count</span></p></td>
<td class="cl-d40c2904"><p class="cl-d40bd274"><span class="cl-d40442a2">15,520 (93.2%)</span><span class="cl-d40442a2"></span></p></td>
<td class="cl-d40c290e"><p class="cl-d40bd274"><span class="cl-d40442a2">203 (1.2%)</span><span class="cl-d40442a2"></span></p></td>
<td class="cl-d40c2918"><p class="cl-d40bd274"><span class="cl-d40442a2">15,723 (94.4%)</span><span class="cl-d40442a2"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-d40c2923"><p class="cl-d40bd26a"><span class="cl-d40442a2">Mar. pct</span><span class="cl-d40442ca"> (1)</span></p></td>
<td class="cl-d40c292c"><p class="cl-d40bd274"><span class="cl-d40442a2"></span><span class="cl-d40442a2">94.8% ; 98.7%</span></p></td>
<td class="cl-d40c2936"><p class="cl-d40bd274"><span class="cl-d40442a2"></span><span class="cl-d40442a2">72.0% ; 1.3%</span></p></td>
<td class="cl-d40c294a"><p class="cl-d40bd274"><span class="cl-d40442a2"></span><span class="cl-d40442a2"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td rowspan="2" class="cl-d40c295e"><p class="cl-d40bd26a"><span class="cl-d40442a2">1</span></p></td>
<td class="cl-d40c28f0"><p class="cl-d40bd26a"><span class="cl-d40442a2">Count</span></p></td>
<td class="cl-d40c2904"><p class="cl-d40bd274"><span class="cl-d40442a2">855 (5.1%)</span><span class="cl-d40442a2"></span></p></td>
<td class="cl-d40c290e"><p class="cl-d40bd274"><span class="cl-d40442a2">79 (0.5%)</span><span class="cl-d40442a2"></span></p></td>
<td class="cl-d40c2918"><p class="cl-d40bd274"><span class="cl-d40442a2">934 (5.6%)</span><span class="cl-d40442a2"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-d40c2972"><p class="cl-d40bd26a"><span class="cl-d40442a2">Mar. pct</span></p></td>
<td class="cl-d40c297c"><p class="cl-d40bd274"><span class="cl-d40442a2"></span><span class="cl-d40442a2">5.2% ; 91.5%</span></p></td>
<td class="cl-d40c2990"><p class="cl-d40bd274"><span class="cl-d40442a2"></span><span class="cl-d40442a2">28.0% ; 8.5%</span></p></td>
<td class="cl-d40c2991"><p class="cl-d40bd274"><span class="cl-d40442a2"></span><span class="cl-d40442a2"></span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-d40c299a"><p class="cl-d40bd26a"><span class="cl-d40442a2">Total</span></p></td>
<td class="cl-d40c29a4"><p class="cl-d40bd26a"><span class="cl-d40442a2">Count</span></p></td>
<td class="cl-d40c29a5"><p class="cl-d40bd274"><span class="cl-d40442a2">16,375 (98.3%)</span><span class="cl-d40442a2"></span></p></td>
<td class="cl-d40c29b8"><p class="cl-d40bd274"><span class="cl-d40442a2">282 (1.7%)</span><span class="cl-d40442a2"></span></p></td>
<td class="cl-d40c29cc"><p class="cl-d40bd274"><span class="cl-d40442a2">16,657 (100.0%)</span><span class="cl-d40442a2"></span></p></td>
</tr>
</tbody>
<tfoot><tr style="overflow-wrap:break-word;"><td colspan="5" class="cl-d40c29d6"><p class="cl-d40bd27e"><span class="cl-d40442ca"> (1)</span><span class="cl-d40442a2"> Columns and rows percentages</span></p></td></tr></tfoot>
</table>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-tbl" id="tbl-cardiovascular-diseases-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) MCASGLOBAL_PRE_M vs AVCGLOBAL_PRE_M
</figcaption></figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-tbl" id="tbl-cardiovascular-diseases-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1.6: Cardiovascular disease events
</figcaption></figure>
</div>
<p>Finally, we save the updated phenotype dataset as a Tab-separated file, using <code><a href="https://readr.tidyverse.org/reference/write_delim.html">readr::write_tsv</a></code>, see <a href="#lst-cartagene-plink2-prep" class="quarto-xref">Listing&nbsp;<span>1.4</span></a> for details:</p>
<div class="cell" data-layout-align="center">
<div id="lst-cartagene-plink2-prep" class="r cell-code listing quarto-float quarto-figure quarto-figure-left anchored">
<figure class="quarto-float quarto-float-lst figure"><figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-cartagene-plink2-prep-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;1.4: Turns out that newest versions of <code>plink2</code> impose explicit encoding of categorical variables, denoting as 1 ‘controls’, and 2 as ‘case’.
</figcaption><div aria-describedby="lst-cartagene-plink2-prep-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode" id="lst-cartagene-plink2-prep"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">## define Boolean categorical variables</span></span>
<span><span class="va">categorical_variables</span> <span class="op">&lt;-</span> <span class="va">cartagene_response_variables</span><span class="op">$</span><span class="va">Features</span><span class="op">[</span><span class="va">cartagene_response_variables</span><span class="op">$</span><span class="va">Type</span><span class="op">==</span><span class="st">"Boolean"</span><span class="op">]</span></span>
<span></span>
<span><span class="va">cartagene_phenodata</span> <span class="op">&lt;-</span> <span class="va">cartagene_phenodata</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/across.html">across</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/all_of.html">all_of</a></span><span class="op">(</span><span class="va">categorical_variables</span><span class="op">)</span>,</span>
<span>                              \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/if_else.html">if_else</a></span><span class="op">(</span><span class="va">x</span> <span class="op">==</span> <span class="fl">0L</span>, <span class="fl">1L</span>, <span class="fl">2L</span>, missing<span class="op">=</span><span class="cn">NA</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/write_delim.html">write_tsv</a></span><span class="op">(</span><span class="va">cartagene_phenodata</span>, </span>
<span>                 file<span class="op">=</span><span class="st">"data/phenotypes/merge_phenos_PCs.txt"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## cartagene_phenodata &lt;- readr::read_tsv("data/phenotypes/merge_phenos_PCs.txt")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
</section></section></section><section id="sec-variant-extraction" class="level3" data-number="1.3.4"><h3 data-number="1.3.4" class="anchored" data-anchor-id="sec-variant-extraction">
<span class="header-section-number">1.3.4</span> Step 2: Variants extraction</h3>
<p>All the curated <code>VCF</code> files have been downloaded and processed by <a href="mailto:access@cartagene.qc.ca">Email Cartagene</a>. Preprocessing details are reported <a href="https://cartagene.qc.ca/files/documents/other/Info_GeneticData3juillet2023.pdf">here</a><a href="#fn5" class="footnote-ref" id="fnref5" role="doc-noteref"><sup>5</sup></a></p>
<section id="sec-HDAC9" class="level4" data-number="1.3.4.1"><h4 data-number="1.3.4.1" class="anchored" data-anchor-id="sec-HDAC9">
<span class="header-section-number">1.3.4.1</span> a) HDAC-family</h4>
<section id="i-retrieve-hdac-positions" class="level5" data-number="1.3.4.1.1"><h5 data-number="1.3.4.1.1" class="anchored" data-anchor-id="i-retrieve-hdac-positions">
<span class="header-section-number">1.3.4.1.1</span> i) Retrieve HDAC positions</h5>
<ul>
<li><p><strong>Objective</strong>: extract the annotated VCF, for the 6 HDAC genes identified of interest, on chromosome 7 (see <a href="#sec-HDAC9" class="quarto-xref"><span>Section 1.3.4.1</span></a> for Bash commands).</p></li>
<li><p><strong>Methods</strong>:</p></li>
</ul>
<ol type="1">
<li>We retrieve positions of the <code>HDAC</code> genes (HDAC 4, 5, 6, 7, 9 and 10) on the latest <code>Hg38</code> Genome Build with script <a href="#lst-hdac-positions" class="quarto-xref">Listing&nbsp;<span>1.5</span></a>. HDAC positions are reported in <a href="#tbl-hdac-positions" class="quarto-xref">Table&nbsp;<span>1.7</span></a><a href="#fn6" class="footnote-ref" id="fnref6" role="doc-noteref"><sup>6</sup></a>
</li>
<li>When several start and end positions were reported for the same gene, we consider the overall <code>min</code> starting position, and the <code>max</code> end position, respectively.</li>
</ol>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">txdb</span> <span class="op">&lt;-</span> <span class="fu">GenomicFeatures</span><span class="fu">::</span><span class="fu">makeTxDbFromUCSC</span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"hg38"</span>, tablename <span class="op">=</span> <span class="st">"refGene"</span><span class="op">)</span></span>
<span><span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">saveDb</a></span><span class="op">(</span><span class="va">txdb</span>, <span class="st">"./data/genome_builds/human_gencode_v42.sqlite"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<div id="lst-hdac-positions" class="r cell-code listing quarto-float quarto-figure quarto-figure-left anchored">
<figure class="quarto-float quarto-float-lst figure"><figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-hdac-positions-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;1.5: Use <code>AnnotationDBI</code> to fetch and retrieve automatically start and end positions of the HDAC family.
</figcaption><div aria-describedby="lst-hdac-positions-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode" id="lst-hdac-positions"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">hdac_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"HDAC4"</span>, <span class="st">"HDAC5"</span>, <span class="st">"HDAC6"</span>, <span class="st">"HDAC7"</span>, <span class="st">"HDAC9"</span>, <span class="st">"HDAC10"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Convert HGNC symbols to Entrez IDs with org.Hs.eg.db</span></span>
<span><span class="co">## 1-1 mapping, great!!</span></span>
<span><span class="va">hdac_entrez_ids</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">mapIds</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keys <span class="op">=</span> <span class="va">hdac_genes</span>,</span>
<span>                     column <span class="op">=</span> <span class="st">"ENTREZID"</span>, keytype <span class="op">=</span> <span class="st">"SYMBOL"</span>, multiVals <span class="op">=</span> <span class="st">"first"</span><span class="op">)</span></span>
<span><span class="va">hdac_entrez_ids</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html">tibble</a></span><span class="op">(</span></span>
<span>  HGNC_SYMBOL <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">hdac_entrez_ids</span><span class="op">)</span>,</span>
<span>  GENEID <span class="op">=</span> <span class="va">hdac_entrez_ids</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## retrieve start and end positions of hdac chromosoms</span></span>
<span><span class="va">txdb</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">loadDb</a></span><span class="op">(</span><span class="st">"data/genome_builds/human_gencode_v42.sqlite"</span><span class="op">)</span></span>
<span><span class="va">hdac_coords</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">select</a></span><span class="op">(</span><span class="va">txdb</span>,</span>
<span>                     keys <span class="op">=</span> <span class="va">hdac_entrez_ids</span><span class="op">$</span><span class="va">GENEID</span>,</span>
<span>                     columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TXCHROM"</span>, <span class="st">"TXSTART"</span>, <span class="st">"TXEND"</span><span class="op">)</span>,</span>
<span>                     keytype <span class="op">=</span> <span class="st">"GENEID"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## take the min and max positions for all chromosomes</span></span>
<span><span class="va">hdac_coords</span> <span class="op">&lt;-</span> <span class="va">hdac_coords</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">GENEID</span>, <span class="va">TXCHROM</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>TXSTART<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">TXSTART</span><span class="op">)</span>, TXEND<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">TXEND</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">ungroup</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">inner_join</a></span><span class="op">(</span><span class="va">hdac_entrez_ids</span>, by <span class="op">=</span> <span class="st">"GENEID"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">GENEID</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/relocate.html">relocate</a></span><span class="op">(</span><span class="va">HGNC_SYMBOL</span><span class="op">)</span></span>
<span><span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/write_delim.html">write_csv</a></span><span class="op">(</span><span class="va">hdac_coords</span>,</span>
<span>                 <span class="st">"data/gene_positions/hdac_coords.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## hdac_coords &lt;- readr::read_csv("data/gene_positions/hdac_coords.csv", </span></span>
<span><span class="co">##                               show_col_types = FALSE)</span></span>
<span><span class="fu"><a href="https://davidgohel.github.io/flextable/reference/flextable.html">flextable</a></span><span class="op">(</span><span class="va">hdac_coords</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/bold.html">bold</a></span><span class="op">(</span>bold <span class="op">=</span> <span class="cn">TRUE</span>, part <span class="op">=</span> <span class="st">"header"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
<div id="tbl-hdac-positions" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-layout-align="center">
<figure class="quarto-float quarto-float-tbl figure"><div aria-describedby="tbl-hdac-positions-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<div class="tabwid">
<style>.cl-e99314cc{}.cl-e980ffee{font-family:'DejaVu Sans';font-size:11pt;font-weight:bold;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-e9810020{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-e9886234{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-e9886248{margin:0;text-align:right;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-e988b3e2{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-e988b3f6{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-e988b3f7{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-e988b40a{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-e988b414{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-e988b428{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style>
<table data-quarto-disable-processing="true" class="cl-e99314cc">
<thead><tr style="overflow-wrap:break-word;">
<th class="cl-e988b3e2"><p class="cl-e9886234"><span class="cl-e980ffee">HGNC_SYMBOL</span></p></th>
<th class="cl-e988b3e2"><p class="cl-e9886234"><span class="cl-e980ffee">GENEID</span></p></th>
<th class="cl-e988b3e2"><p class="cl-e9886234"><span class="cl-e980ffee">TXCHROM</span></p></th>
<th class="cl-e988b3f6"><p class="cl-e9886248"><span class="cl-e980ffee">TXSTART</span></p></th>
<th class="cl-e988b3f6"><p class="cl-e9886248"><span class="cl-e980ffee">TXEND</span></p></th>
</tr></thead>
<tbody>
<tr style="overflow-wrap:break-word;">
<td class="cl-e988b3f7"><p class="cl-e9886234"><span class="cl-e9810020">HDAC6</span></p></td>
<td class="cl-e988b3f7"><p class="cl-e9886234"><span class="cl-e9810020">10013</span></p></td>
<td class="cl-e988b3f7"><p class="cl-e9886234"><span class="cl-e9810020">chrX</span></p></td>
<td class="cl-e988b40a"><p class="cl-e9886248"><span class="cl-e9810020">48,801,398</span></p></td>
<td class="cl-e988b40a"><p class="cl-e9886248"><span class="cl-e9810020">48,824,982</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-e988b3f7"><p class="cl-e9886234"><span class="cl-e9810020">HDAC5</span></p></td>
<td class="cl-e988b3f7"><p class="cl-e9886234"><span class="cl-e9810020">10014</span></p></td>
<td class="cl-e988b3f7"><p class="cl-e9886234"><span class="cl-e9810020">chr17</span></p></td>
<td class="cl-e988b40a"><p class="cl-e9886248"><span class="cl-e9810020">44,076,753</span></p></td>
<td class="cl-e988b40a"><p class="cl-e9886248"><span class="cl-e9810020">44,123,641</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-e988b3f7"><p class="cl-e9886234"><span class="cl-e9810020">HDAC7</span></p></td>
<td class="cl-e988b3f7"><p class="cl-e9886234"><span class="cl-e9810020">51564</span></p></td>
<td class="cl-e988b3f7"><p class="cl-e9886234"><span class="cl-e9810020">chr12</span></p></td>
<td class="cl-e988b40a"><p class="cl-e9886248"><span class="cl-e9810020">47,782,724</span></p></td>
<td class="cl-e988b40a"><p class="cl-e9886248"><span class="cl-e9810020">47,819,903</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-e988b3f7"><p class="cl-e9886234"><span class="cl-e9810020">HDAC10</span></p></td>
<td class="cl-e988b3f7"><p class="cl-e9886234"><span class="cl-e9810020">83933</span></p></td>
<td class="cl-e988b3f7"><p class="cl-e9886234"><span class="cl-e9810020">chr22</span></p></td>
<td class="cl-e988b40a"><p class="cl-e9886248"><span class="cl-e9810020">50,245,184</span></p></td>
<td class="cl-e988b40a"><p class="cl-e9886248"><span class="cl-e9810020">50,251,265</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-e988b3f7"><p class="cl-e9886234"><span class="cl-e9810020">HDAC9</span></p></td>
<td class="cl-e988b3f7"><p class="cl-e9886234"><span class="cl-e9810020">9734</span></p></td>
<td class="cl-e988b3f7"><p class="cl-e9886234"><span class="cl-e9810020">chr7</span></p></td>
<td class="cl-e988b40a"><p class="cl-e9886248"><span class="cl-e9810020">18,086,825</span></p></td>
<td class="cl-e988b40a"><p class="cl-e9886248"><span class="cl-e9810020">19,002,414</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-e988b414"><p class="cl-e9886234"><span class="cl-e9810020">HDAC4</span></p></td>
<td class="cl-e988b414"><p class="cl-e9886234"><span class="cl-e9810020">9759</span></p></td>
<td class="cl-e988b414"><p class="cl-e9886234"><span class="cl-e9810020">chr2</span></p></td>
<td class="cl-e988b428"><p class="cl-e9886248"><span class="cl-e9810020">239,048,168</span></p></td>
<td class="cl-e988b428"><p class="cl-e9886248"><span class="cl-e9810020">239,401,649</span></p></td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-tbl" id="tbl-hdac-positions-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1.7: HDAC Gene Positions
</figcaption></figure>
</div>
</div>
<ul>
<li><span data-fg="red" style="color: red;"><strong>Remark</strong>: Instead of fetching start and end locations automatically, add an <code>INFO/GENE</code> field directly within the <code>VCF</code> files.</span></li>
</ul></section><section id="ii-extract-snps-corresponding-to-provided-hdac-gene" class="level5" data-number="1.3.4.1.2"><h5 data-number="1.3.4.1.2" class="anchored" data-anchor-id="ii-extract-snps-corresponding-to-provided-hdac-gene">
<span class="header-section-number">1.3.4.1.2</span> ii) Extract SNPs corresponding to provided HDAC gene</h5>
<ul>
<li>
<strong>Inputs</strong>:
<ul>
<li>Cleaned <strong>VCF</strong> file of the chromosome 7 (where <code>HDAC-9</code> is present)<a href="#fn7" class="footnote-ref" id="fnref7" role="doc-noteref"><sup>7</sup></a>.</li>
</ul>
</li>
<li>
<strong>Outputs</strong>:
<ul>
<li>
<code>chr7.HDAC9.vcf</code>: VCF file, restrained to HDAC9 region with affiliated SNPs.</li>
</ul>
</li>
<li>
<strong>Objective</strong>: Extract annotated SNPs within <code>HDAC9</code> boundaries, as defined by the <code>hg38</code> reference genome. <strong>Details</strong>:
<ul>
<li>Discard variants with <strong>Minor Allele Frequency (MAF)</strong> <span class="math inline">\(&lt; 0.01\)</span> (in other words, the SNP must be present in at least <span class="math inline">\(1\%\)</span> of the samples).</li>
<li>SNPs were extracted using <code>bcftools view</code>, with shell script <a href="./shell/extract_variants.sh">extract_variants.sh</a><a href="#fn8" class="footnote-ref" id="fnref8" role="doc-noteref"><sup>8</sup></a>:</li>
</ul>
</li>
</ul>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nohup</span> ./shell/extract_variants.sh <span class="op">&gt;</span> shell/extract_variants.log <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">&amp;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li>
<strong>Remark:</strong> <code>--maf</code> guarantees removal of highly recessive SNPs, present in less than <span class="math inline">\(1\%\)</span> of cases with respect to the dominant form: infrequent SNPs are indeed associated with lower statistical power.</li>
</ul>
<!-- ##### b) HLA-family {#sec-HLA} --><!-- HLA genes are usually classified into **Class I (HLA-A, HLA-B, HLA-C)** and **Class II (HLA-DRB1, HLA-DQA1, HLA-DQB1)**, amounting in total to six main genes within the human organism. --><!-- To retrieve their positions on the latest `Hg38` Genome Build in an automated fashion, see @lst-hla-positions. HLA positions are reported in @tbl-hla-positions. [Do not use the `BiomarT` package: it's not anymore maintained, and can only be used with out-of-date `Hg37` Human Genome Build. Report to [Genome Builds Versions](https://www.linkedin.com/posts/%F0%9F%8E%AF-ming-tommy-tang-40650014_genome-builds-matter-avoid-costly-mistakes-activity-7293270985930141698-ZFkH) for details. Besides, note that when several start and end positions were assigned to the same gene, we consider each time the minimal starting position, and the maximal end position.]{fg="red"} --><!-- ```{r} --><!-- #| label: tbl-hla-positions --><!-- #| lst-label: lst-hla-positions --><!-- #| lst-cap:  Use `AnnotationDBI` to fetch and retrieve automatically start and end positions of the HLA family.  --><!-- #| tbl-cap: "HLA Gene Positions" --><!-- ## renv::install(packages = c("bioc::AnnotationDbi", "bioc::org.Hs.eg.db")) --><!-- ## retrieve latest human genome assembly --><!-- ## hla_genes <- c("HLA-A", "HLA-B", "HLA-C", "HLA-DQA1", "HLA-DQB1", "HLA-DRB1") --><!-- ## ## Convert HGNC symbols to Entrez IDs with org.Hs.eg.db --><!-- ## ## 1-1 mapping, great!! --><!-- ## hla_entrez_ids <- AnnotationDbi::mapIds(org.Hs.eg.db, keys = hla_genes,  --><!-- ##                      column = "ENTREZID", keytype = "SYMBOL", multiVals = "first") --><!-- ## hla_entrez_ids <- tibble::tibble( --><!-- ##   HGNC_SYMBOL = names(hla_entrez_ids), --><!-- ##   GENEID = hla_entrez_ids) --><!-- ##  --><!-- ## ## retrieve start and end positions of HLA chromosoms --><!-- ## txdb <- AnnotationDbi::loadDb("data/genome_builds/txdb_human_gencode_v42.sqlite") --><!-- ## hla_coords <- AnnotationDbi::select(txdb,  --><!-- ##                      keys = hla_entrez_ids$GENEID,  --><!-- ##                      columns = c("TXCHROM", "TXSTART", "TXEND"),  --><!-- ##                      keytype = "GENEID") --><!-- ##  --><!-- ## ## take the min and max positions for all chromosomes --><!-- ## hla_coords <- hla_coords |>  --><!-- ##   dplyr::filter(TXCHROM=="chr6") |>  --><!-- ##   dplyr::group_by(GENEID, TXCHROM) |>  --><!-- ##   dplyr::summarise(TXSTART=min(TXSTART), TXEND=max(TXEND)) |>  --><!-- ##   dplyr::ungroup() |>  --><!-- ##   dplyr::inner_join(hla_entrez_ids, by = "GENEID") |>  --><!-- ##   dplyr::arrange(GENEID) |>  --><!-- ##   relocate(HGNC_SYMBOL) --><!-- ## readr::write_csv(hla_coords,  --><!-- ##                  "data/gene_positions/hla_coords.csv") --><!-- hla_coords <- readr::read_csv("data/gene_positions/hla_coords.csv",  --><!--                               show_col_types = FALSE) --><!-- flextable(hla_coords) |>  --><!--   bold(bold = TRUE, part = "header") --><!-- ``` --></section></section></section><section id="step-3-gwas-analyses" class="level3" data-number="1.3.5"><h3 data-number="1.3.5" class="anchored" data-anchor-id="step-3-gwas-analyses">
<span class="header-section-number">1.3.5</span> Step 3: GWAS Analyses</h3>
<section id="sec-GWAS" class="level4" data-number="1.3.5.1"><h4 data-number="1.3.5.1" class="anchored" data-anchor-id="sec-GWAS">
<span class="header-section-number">1.3.5.1</span> i) GLM and GWAS</h4>
<ul>
<li><p><strong>Objective</strong>: Use of <code>PLINK2 + glm</code> with the first 10 principal components as covariates, see <a href="#lst-glm-bash" class="quarto-xref">Listing&nbsp;<span>1.6</span></a>, using logistic regression for categorical variables, and <code>lm</code> for continuous outcomes.</p></li>
<li>
<p><strong>Input</strong>:</p>
<ul>
<li>
<code>VCF</code> file on the region/gene of interest</li>
<li>Phenotypes, with individual patient IDs.</li>
<li>Explanatory variable to predict, provided with <code>--pheno-name &lt;response_variable&gt;</code>.</li>
</ul>
</li>
</ul>
<div id="lst-glm-bash" class="listing quarto-float quarto-figure quarto-figure-left anchored">
<figure class="quarto-float quarto-float-lst figure"><figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst quarto-uncaptioned" id="lst-glm-bash-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;1.6
</figcaption><div aria-describedby="lst-glm-bash-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<p><strong>Template GLM instruction for GWAS studies:</strong></p>
<div class="sourceCode" id="annotated-cell-24"><pre class="sourceCode bash code-annotation-code code-with-copy code-annotated"><code class="sourceCode bash"><span id="annotated-cell-24-1"><a href="#annotated-cell-24-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="annotated-cell-24-2"><a href="#annotated-cell-24-2" aria-hidden="true" tabindex="-1"></a><span class="ex">plink2</span> <span class="dt">\</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-24" data-target-annotation="1">1</button><span id="annotated-cell-24-3" class="code-annotation-target"><a href="#annotated-cell-24-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--double-id</span> <span class="dt">\ </span>#</span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-24" data-target-annotation="2">2</button><span id="annotated-cell-24-4" class="code-annotation-target"><a href="#annotated-cell-24-4" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--pheno</span> ./phenotypes/merge_phenos_PCs.txt <span class="dt">\ </span>#</span>
<span id="annotated-cell-24-5"><a href="#annotated-cell-24-5" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--pheno-name</span> OSTEOPONIA <span class="dt">\ </span>#</span>
<span id="annotated-cell-24-6"><a href="#annotated-cell-24-6" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--vcf</span> /mnt/projects_tn01/Cartagene/analyses/variants_extraction/chr7.merged.clean.noMono.extracted_variants.HDAC9.vcf dosage=HDS <span class="co">#</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-24" data-target-annotation="3">3</button><span id="annotated-cell-24-7" class="code-annotation-target"><a href="#annotated-cell-24-7" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--glm</span> hide-covar <span class="dt">\ </span>#</span>
<span id="annotated-cell-24-8"><a href="#annotated-cell-24-8" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--covar</span> /mnt/projects_tn01/Cartagene/analyses/phenotypes/merge_phenos_PCs.txt <span class="dt">\ </span>#</span>
<span id="annotated-cell-24-9"><a href="#annotated-cell-24-9" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--covar-name</span> PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10 <span class="dt">\ </span>#</span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-24" data-target-annotation="4">4</button><span id="annotated-cell-24-10" class="code-annotation-target"><a href="#annotated-cell-24-10" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--out</span> HDAC9_OSTEOPONIA <span class="co">#</span></span><div class="code-annotation-gutter-bg"></div><div class="code-annotation-gutter"></div></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<dl class="code-annotation-container-hidden code-annotation-container-grid">
<dt data-target-cell="annotated-cell-24" data-target-annotation="1">1</dt>
<dd>
<span data-code-cell="annotated-cell-24" data-code-lines="3" data-code-annotation="1">The <code>--double-id</code> option ensures independent GWAS analyses per individual, aka the <em>single-sample mode</em>, stating explicitly that the <code>VCF</code> file follows a <em>family-based format</em> (FID/IID pairs).</span>
</dd>
<dt data-target-cell="annotated-cell-24" data-target-annotation="2">2</dt>
<dd>
<span data-code-cell="annotated-cell-24" data-code-lines="4,5,6" data-code-annotation="2"><strong>GWAS inputs</strong>: We need the <em>phenotype information</em> (provided with <code>--pheno</code> and <code>--keep</code> commands), the <em>variable to predict</em> (provided with <code>--pheno-name</code>) and the <em>VCF file</em> (command <code>--vcf</code>), here using the SNPs annotations for the HDAC9 gene<a href="#fn9" class="footnote-ref" id="fnref9" role="doc-noteref"><sup>9</sup></a>.</span>
</dd>
<dt data-target-cell="annotated-cell-24" data-target-annotation="3">3</dt>
<dd>
<span data-code-cell="annotated-cell-24" data-code-lines="7,8,9" data-code-annotation="3">GWAS model options: <code>--glm</code> is the general linear model, which uses by default a <em>logistic regression</em> for categorical variables, and a <em>standard linear Gaussian model</em>, equivalent to <code>lm</code> for continuous variables<a href="#fn10" class="footnote-ref" id="fnref10" role="doc-noteref"><sup>10</sup></a>. <code>--covar</code> is the covariate file, with relevant explanatory variables to integrate reported with <code>--covar-name</code> (see @ for details). PCAs are used to describe the population structure in an unsupervised manner; and avoid and detect latent subgroups.</span>
</dd>
<dt data-target-cell="annotated-cell-24" data-target-annotation="4">4</dt>
<dd>
<span data-code-cell="annotated-cell-24" data-code-lines="10" data-code-annotation="4">The output GWAS folder, with <code>--out</code> command. Stored for now in <code>/mnt/projects_tn01/Cartagene/analyses/association</code>.</span>
</dd>
</dl>
</div>
</figure>
</div>
<p>In practice, run the <a href="./shell/plink2_gwas.sh"><code>plink2_gwas.sh</code></a> script in the background (possibility to customise gene name and feature variables to regress on):</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co">## Run the GWAS pipeline in the background,</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="co">## persist after logout, and log everything.</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="fu">nohup</span> ./shell/plink2_gwas.sh <span class="op">&gt;</span> shell/plink2_gwas.log <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">&amp;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li><p><strong>Remark 1</strong>: you may come up with <code>Error: Cannot proceed with --glm regression on phenotype 'TACAIX', since variance inflation factor for covariate 'PC2' is too high</code>. In this case, you may try removing completely covariates (with <code>--glm allow-no-covars</code>), or/and increase variance inflation threshold (<code>--vif number_vif</code> option)<a href="#fn11" class="footnote-ref" id="fnref11" role="doc-noteref"><sup>11</sup></a></p></li>
<li><p><span data-fg="red" style="color: red;"><strong>Remark 2</strong>: All the pre-processing operations detailed in <a href="#sec-preprocessing-redundancy" class="quarto-xref"><span>Section 1.3.2.2</span></a> are only used for the computation of the PCA components, but not subsequently used in the <code>glm</code> regression for trimming strongly correlated features.</span></p></li>
<li><p><span data-fg="red" style="color: red;"><strong>Remark 3</strong>: Add phenotype covariates, such as <code>SEX</code> or <code>AGE</code> in the regression framework, which can play a strong leverage on the impact of SNPs. <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/GENESIS/inst/doc/assoc_test.html#mixed-model-association-testing"><code>GENESIS</code></a> is a R package mixing environmental and genetic factors in a <em>fixed linear model</em> approach.</span></p></li>
</ul></section><section id="ii-gwas-visualisations" class="level4" data-number="1.3.5.2"><h4 data-number="1.3.5.2" class="anchored" data-anchor-id="ii-gwas-visualisations">
<span class="header-section-number">1.3.5.2</span> ii) GWAS Visualisations</h4>
<p>For each gene-phenotype pair, we generate cognate <strong>Manhattan plots</strong><a href="#fn12" class="footnote-ref" id="fnref12" role="doc-noteref"><sup>12</sup></a>, and <span class="math inline">\(p\)</span>-values distributions, using <em>histograms</em><a href="#fn13" class="footnote-ref" id="fnref13" role="doc-noteref"><sup>13</sup></a>, and <strong>QQplots</strong><a href="#fn14" class="footnote-ref" id="fnref14" role="doc-noteref"><sup>14</sup></a>. Core function for generating these 3 GWAS visualisations is <a href="./R/gwas_plots.R"><code>generate_gwas_per_phenotype</code></a>. An example for gene <code>HDAC-9</code> is provided in <a href="#fig-GWAS-HDAC9" class="quarto-xref">Figure&nbsp;<span>1.3</span></a>.</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">## p-value adjustment</span></span>
<span><span class="co">## genename &lt;- "HDAC-9"; pheno_label &lt;- "OSTEOPONIA"; filedate &lt;- "2025-04-07"</span></span>
<span><span class="va">num_variants</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_csv</a></span><span class="op">(</span><span class="st">"./tables/HDAC-9/HDAC-9_AVCGLOBAL_PRE_M_2025-04-07.AVCGLOBAL_PRE_M.glm.logistic.hybrid"</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">pval_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.05</span><span class="op">/</span><span class="op">(</span><span class="va">num_variants</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">cartagene_response_variables</span><span class="op">$</span><span class="va">Features</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## pval_threshold &lt;- 0.01/num_variants</span></span>
<span></span>
<span><span class="va">gwas_plots_hdac</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">cartagene_response_variables</span><span class="op">$</span><span class="va">Features</span>, </span>
<span>                          <span class="kw">function</span><span class="op">(</span><span class="va">feature_label</span><span class="op">)</span> <span class="fu">generate_gwas_per_phenotype</span> <span class="op">(</span>genename <span class="op">=</span> <span class="st">"HDAC-9"</span>,</span>
<span>                                                                               pheno_label <span class="op">=</span> <span class="va">feature_label</span>, filedate<span class="op">=</span><span class="st">"2025-04-07"</span>, pval_threshold <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">gwas_plots_hdac</span> <span class="op">&lt;-</span> <span class="fu">gridExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html">marrangeGrob</a></span><span class="op">(</span><span class="va">gwas_plots_hdac</span>, nrow<span class="op">=</span><span class="fl">1</span>, ncol<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"figures/gwas_HDAC9.pdf"</span>,</span>
<span>       <span class="va">gwas_plots_hdac</span>, dpi <span class="op">=</span> <span class="fl">600</span>,</span>
<span>       width <span class="op">=</span> <span class="fl">8</span>, height <span class="op">=</span> <span class="fl">12</span><span class="op">)</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="fig-GWAS-HDAC9" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-GWAS-HDAC9-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<p><object data="./figures/gwas_HDAC9.pdf" type="application/pdf" width="80%" height="800"><a href="./figures/gwas_HDAC9.pdf" download="">Download PDF file.</a></object></p>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-GWAS-HDAC9-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1.3
</figcaption></figure>
</div>
</section><section id="iii-multiple-test-correction" class="level4" data-number="1.3.5.3"><h4 data-number="1.3.5.3" class="anchored" data-anchor-id="iii-multiple-test-correction">
<span class="header-section-number">1.3.5.3</span> iii) Multiple test correction</h4>
<p>To compute the adjusted <em>significance threshold</em>, we applied a <code>FWER</code>-like approach, conventional in the GWAS field: <code>{r} 0.05/(num_variants*length(cartagene_response_variables$Features))</code>, in which you divide the pre-defined threshold (standard: <code>0.05</code>) by the total number of pairwise tests carried out (number of variants times number of regression variables times). Final threshold is accordingly: 2.1188236^{-6}, as 2622 variants were extracted, and 9 phenotype variables were predicted.</p>
<p><span data-fg="red" style="color: red;">The FWER approach is particularly <strong>conservative</strong>, in other words, only the most significant differences are detected.</span></p>
<ul>
<li>
<p><strong>Perspectives</strong>:</p>
<ul>
<li>
<a href="https://www.linkedin.com/posts/adrianolszewski_statistic-datascience-research-activity-7301605962820284417-ho3Z">Recent Adjusted <span class="math inline">\(p\)</span>-value correction method</a>.</li>
<li>
<a href="https://www.taylorfrancis.com/books/mono/10.1201/9781420010909/multiple-comparisons-using-frank-bretz-torsten-hothorn-peter-westfall">Multiple Comparisons Using R</a>, comprehensive resource of distinct R package strategies for correcting for multiplicity of <span class="math inline">\(p\)</span>-values, from <span class="citation" data-cites="bretz2016">Bretz, Hothorn, and Westfall (<a href="references.html#ref-bretz2016" role="doc-biblioref">2016</a>)</span>. Report also to <a href="#nte-MTC" class="quarto-xref">Note&nbsp;<span>1.1</span></a> for an overview of existing approaches.</li>
</ul>
</li>
</ul>
<div id="nte-MTC" class="callout callout-style-default callout-note callout-titled" title="Classes of Multiple Test correction approaches">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-2-contents" aria-controls="callout-2" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note&nbsp;1.1: Classes of Multiple Test correction approaches
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-2" class="callout-2-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p><strong>Multiple comparison procedures</strong> (MCPs) can be classified into the following adjustment methods:</p>
<ol type="1">
<li><p><em>Fixed sequential</em>: no adjustment as long as H0s are rejected. One non-rejection and it stops testing with all others non-rejected.</p></li>
<li><p><em>Callback</em>: significance level (e.g.&nbsp;0.05) is split between all hypotheses (can be further weighted). With rejection of the subsequent H0s, their fractions of significance level is accumulated and passed to the next comparison.</p></li>
<li><p><em>Gatekeeper</em>: serial, parallel &amp; combined trees, collecting hypotheses into families. Groups of testing can be interpreted as <em>gatekeepers</em>: only if any <span class="math inline">\(H_0\)</span> in a family is rejected that the significance is propagated to the next family. The approach is the most versatile, and can be flexibly implemented in R package <a href="https://merck.github.io/gMCPLite"><code>gMCPLite</code></a>. See also <a href="https://baselbiometrics.github.io/home/docs/talks/20220329/2_Glimm_Bretz_Xi.pdf">Graphical Approaches to Multiple Test Problems</a>, for additional details.</p></li>
</ol>
</div>
</div>
</div>
</section></section></section><section id="perspectives-aka-post-gwas" class="level2" data-number="1.4"><h2 data-number="1.4" class="anchored" data-anchor-id="perspectives-aka-post-gwas">
<span class="header-section-number">1.4</span> Perspectives, aka post GWAS</h2>
<section id="post-hoc-gwas" class="level3" data-number="1.4.1"><h3 data-number="1.4.1" class="anchored" data-anchor-id="post-hoc-gwas">
<span class="header-section-number">1.4.1</span> Post-hoc GWAS</h3>
<ul>
<li><a href="https://varchat.engenome.com/">Collect information on variants with LLM-like <code>VarChat</code> app</a></li>
</ul></section><section id="explore-non-coding-regulatory-regions-from-wes-to-wgs" class="level3" data-number="1.4.2"><h3 data-number="1.4.2" class="anchored" data-anchor-id="explore-non-coding-regulatory-regions-from-wes-to-wgs">
<span class="header-section-number">1.4.2</span> Explore non-coding, regulatory regions (from WES to WGS)</h3>
<ul>
<li><p>Paper <a href="https://www.nature.com/articles/s41588-023-01529-1">Systematic differences in discovery of genetic effects on gene expression and complex traits</a>, from <span class="citation" data-cites="mostafavi2023ng">Mostafavi et al. (<a href="references.html#ref-mostafavi2023ng" role="doc-biblioref">2023</a>)</span>, suggest extending extraction of genomic regions of interest beyond coding sequences (report to <a href="#sec-variant-extraction" class="quarto-xref"><span>Section 1.3.4</span></a> for current approach). Indeed, the article demonstrates that most of identified signals originate from nearby gene regulatory sites.</p></li>
<li><p><a href="https://www.biorxiv.org/content/10.1101/2024.08.01.606099v1"><code>EPInformer</code>: a scalable deep learning framework for gene expression prediction by integrating promoter-enhancer sequences with multimodal epigenomic data</a>, from <span class="citation" data-cites="lin2024">Lin, Luo, and Pinello (<a href="references.html#ref-lin2024" role="doc-biblioref">2024</a>)</span>. Avalaible as a <a href="https://github.com/pinellolab/EPInformer">GH package</a>. Can infer and predict gene expression from promoter and enhancer sequences paired with epigenomic signals; identify the most significant enhancers; and identify <em>regulatory sequences</em> and <em>transcription factor binding motifs</em>.</p></li>
</ul></section><section id="post-gwas-explore-other-types-of-genetic-variations" class="level3" data-number="1.4.3"><h3 data-number="1.4.3" class="anchored" data-anchor-id="post-gwas-explore-other-types-of-genetic-variations">
<span class="header-section-number">1.4.3</span> Post-GWAS: Explore other types of genetic variations</h3>
<ul>
<li><p><a href="https://www.nature.com/articles/s41576-024-00778-y">Genome-wide association testing beyond SNPs</a>, from <span class="citation" data-cites="harris2025nrg">Harris et al. (<a href="references.html#ref-harris2025nrg" role="doc-biblioref">2025</a>)</span></p></li>
<li><p><a href="https://www.frontiersin.org/journals/genetics/articles/10.3389/fgene.2022.1006903/full">Benchmarking post-GWAS analysis tools in major depression: Challenges and implications</a>, from <span class="citation" data-cites="perez-granado2022fg">Pérez-Granado, Piñero, and Furlong (<a href="references.html#ref-perez-granado2022fg" role="doc-biblioref">2022</a>)</span>.</p></li>
</ul></section><section id="sec-post-GWAS" class="level3" data-number="1.4.4"><h3 data-number="1.4.4" class="anchored" data-anchor-id="sec-post-GWAS">
<span class="header-section-number">1.4.4</span> Post-GWAS: Retrieve automatically variants of interest</h3>
<p>This analysis is not formally a GWAS study, but rather a <em>gene-centric</em> study, pre-selecting genes based on prior knowledge and/or intuition.</p>
<p>Alternatively, we could consider a data-driven, and more agnostic approach, to refine the selection of gene candidates. Several approaches to that end have been implemented:</p>
<ul>
<li><p><span class="citation" data-cites="gnanaolivu2025bb">Gnanaolivu et al. (<a href="references.html#ref-gnanaolivu2025bb" role="doc-biblioref">2025</a>)</span> proposed in <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-025-06096-2">A clinical knowledge graph-based framework to prioritize candidate genes for facilitating diagnosis of Mendelian diseases and rare genetic conditions</a>, the <code>phenotype prioritization and analysis for rare diseases, PPAR</code> algorithm to rank genes based on <em>human phenotype ontology (HPO)</em> terms.<a href="#fn15" class="footnote-ref" id="fnref15" role="doc-noteref"><sup>15</sup></a>. In details, <code>PPAR</code> combines embeddings from the <em>human knowledge graph</em>, incorporating genes, HPO terms, and gene ontology annotations connections. For each input HPO term, a prioritized list of genes is returned based on their relevance and similarity to the HPO term.</p></li>
<li><p><span class="citation" data-cites="bridges2025bb">Bridges et al. (<a href="references.html#ref-bridges2025bb" role="doc-biblioref">2025</a>)</span> developed in <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-025-06105-4">Towards a standard benchmark for phenotype-driven variant and gene prioritisation algorithms: <code>PhEval</code>, Phenotypic inference Evaluation framework</a> a unified benchmarking platform to clean inputs for <em>phenotype-driven VGPAs</em>. Variant and Gene Prioritisation Algorithms integrate complex and multi-modal datasets, such as ontologies and gene-to-phenotype associations, to predict the most influential and promising gene targets controlling the evolution of rare diseases. Model is available as GitHub Repo <a href="https://github.com/monarch-initiative/pheval"><code>PhEval</code></a>.</p></li>
</ul></section><section id="post-gwas-incorporate-familial-pedigrees" class="level3" data-number="1.4.5"><h3 data-number="1.4.5" class="anchored" data-anchor-id="post-gwas-incorporate-familial-pedigrees">
<span class="header-section-number">1.4.5</span> Post-GWAS: Incorporate familial pedigrees</h3>
<ul>
<li><p><a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-025-11344-z">Genome-wide association study of fat content and fatty acid composition of shea tree.</a>, from <span class="citation" data-cites="attikora2025bg">Attikora et al. (<a href="references.html#ref-attikora2025bg" role="doc-biblioref">2025</a>)</span>. Under the hood, relies on the <code>mrMLM</code> R package to include both the <em>population structure matrix</em> and the <em>kinship matrix</em>. Provides a complete GWAS pipeline, including a number of meaningful illustrations.</p></li>
<li><p><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-025-06077-5"><code>BHCox</code>: Bayesian heredity-constrained Cox proportional hazards models for detecting gene-environment interactions</a>, from <span class="citation" data-cites="sun2025bb">Sun et al. (<a href="references.html#ref-sun2025bb" role="doc-biblioref">2025</a>)</span>. Under the hood, relies on the <code>brms</code> R package.</p></li>
</ul></section><section id="post-gwas-polygenic-risk-scores" class="level3" data-number="1.4.6"><h3 data-number="1.4.6" class="anchored" data-anchor-id="post-gwas-polygenic-risk-scores">
<span class="header-section-number">1.4.6</span> Post-GWAS: Polygenic risk scores</h3>
<ul>
<li><p><a href="https://doi.org/10.1186/s13059-024-03400-w">Optimizing and benchmarking polygenic risk scores with GWAS summary statistics</a>, from <span class="citation" data-cites="zhao2024gb">Z. Zhao et al. (<a href="references.html#ref-zhao2024gb" role="doc-biblioref">2024</a>)</span>.</p></li>
<li><p><a href="10.1093/bioinformatics/btaf143"><code>XPRS</code>: A Tool for Interpretable and Explainable Polygenic Risk Score</a>, from <span class="citation" data-cites="kim2025b">Kim and Lee (<a href="references.html#ref-kim2025b" role="doc-biblioref">2025</a>)</span>, is a R package for inferring the <em>polygenic risk score</em> for assessing genetic susceptibility to diseases, providing additional interpretation and visualisation tools. PRSs are further split and classified into genes and single nucleotide polymorphism (SNP) contribution scores via <em>Shapley additive explanations</em> (SHAPs), visualised as <strong>Manhattan plots</strong>, <strong>LocusZoom-like</strong> plots and tables at the population and individual levels.</p></li>
<li><p><a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btaf071/8015614"><code>PNL</code>: a software to build polygenic risk scores using a Super Learner approach based on <code>PairNet</code>, a Convolutional Neural Network</a>, from <span class="citation" data-cites="chen2025b">Chen et al. (<a href="references.html#ref-chen2025b" role="doc-biblioref">2025</a>)</span>.</p></li>
</ul></section><section id="post-gwas-orthogonal-integration" class="level3" data-number="1.4.7"><h3 data-number="1.4.7" class="anchored" data-anchor-id="post-gwas-orthogonal-integration">
<span class="header-section-number">1.4.7</span> Post-GWAS: Orthogonal integration</h3>
<section id="epigenomic" class="level4" data-number="1.4.7.1"><h4 data-number="1.4.7.1" class="anchored" data-anchor-id="epigenomic">
<span class="header-section-number">1.4.7.1</span> Epigenomic</h4>
<ul>
<li>
<span class="citation" data-cites="zhao2025bib">G. Zhao and Lai (<a href="references.html#ref-zhao2025bib" role="doc-biblioref">2025</a>)</span> implements <a href="https://academic.oup.com/bib/article/26/2/bbaf123/8092303?searchresult=1&amp;login=false"><code>SC-VAR</code>: a computational tool for interpreting polygenic disease risks using single-cell epigenomic data</a>, a novel computational tool available as a GitHub repo and Python application: <a href="https://github.com/gefeiZ/SC_VAR"><code>SC-VAR</code> GitHub</a>. <code>SC-VAR</code> uses single-cell <em>epigenomic</em> data to predict functional outcomes of the identified disease-associated <em>GWAS variants</em>, enhancing their interpretability. Under the hood, relies on <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004219"><code>MAGMA</code> linear modelling framework</a>, developed by <span class="citation" data-cites="leeuw2015pcb">Leeuw et al. (<a href="references.html#ref-leeuw2015pcb" role="doc-biblioref">2015</a>)</span>.</li>
</ul></section><section id="eqtl-quantitative-expression" class="level4" data-number="1.4.7.2"><h4 data-number="1.4.7.2" class="anchored" data-anchor-id="eqtl-quantitative-expression">
<span class="header-section-number">1.4.7.2</span> eQTL (quantitative expression)</h4>
<ul>
<li>
<a href="https://github.com/statgen/fivex"><code>FIVEx</code></a> is a web-based interface to explore <code>CarTaGene</code> to visualize and query eQTL (quantitative trait loci) data in various ways. <strong>Would need a stringent maintenance!!</strong>
</li>
</ul></section></section></section><section id="appendix" class="level2 appendix" data-number="1.5"><h2 class="appendix anchored" data-number="1.5" data-anchor-id="appendix">
<span class="header-section-number">1.5</span> Appendix</h2>
<section id="sec-DSL-Nextflow" class="level3" data-number="1.5.1"><h3 data-number="1.5.1" class="anchored" data-anchor-id="sec-DSL-Nextflow">
<span class="header-section-number">1.5.1</span> Scalability with <code>Nextflow</code>
</h3>
<p>Lots of intermediate files are not required for downstream analyses, hence it would be relevant to rely on an existing <em>Nextflow</em> or <em>Snakemake</em> DSL workflows:</p>
<ul>
<li><p><a href="https://genepi.github.io/nf-gwas/"><code>nf-GWAS</code> ‘Nextflow pipeline’</a>, from <span class="citation" data-cites="schonherr2024">Schönherr and Forer (<a href="references.html#ref-schonherr2024" role="doc-biblioref">[2021] 2024</a>)</span>, is actively maintained by Curie Bioinformatics Team, and includes the latest <code>plink2</code> facilities.</p></li>
<li><p><a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btaf105/8069456"><code>PopGLen</code> ‘Snakemake’ pipeline</a>, from <span class="citation" data-cites="nolen2025b">Nolen (<a href="references.html#ref-nolen2025b" role="doc-biblioref">2025</a>)</span>. <span data-fg="red" style="color: red;">Not for running GWAS analyses, but rather for evaluating the quality and impact of preprocessing and quality mapping at the population-level genome starting from <code>FASTQ</code> files.</span></p></li>
</ul>
<!-- ### Clean Git History --><!-- ``` bash --><!-- python3 bin/git-filter-repo --path data/variants/chr6.HLA-complex.vcf --invert-paths --><!-- ``` -->

<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-attikora2025bg" class="csl-entry" role="listitem">
Attikora, Affi Jean Paul, Kouakou Alfred Kouassi, Saraka Didier Martial Yao, Dougba Noel Dago, Souleymane Silué, Caroline De Clerck, Nafan Diarrassouba, et al. 2025. <span>‘Genome-Wide Association Study of Fat Content and Fatty Acid Composition of Shea Tree (<span>Vitellaria</span> Paradoxa <span>C</span>.<span>F</span>. <span>Gaertn</span> Subsp. Paradoxa)’</span>. <em>BMC Genomics</em> 26 (1): 164. <a href="https://doi.org/10.1186/s12864-025-11344-z">https://doi.org/10.1186/s12864-025-11344-z</a>.
</div>
<div id="ref-awadalla2013ijoe" class="csl-entry" role="listitem">
Awadalla, Philip, Catherine Boileau, Yves Payette, Youssef Idaghdour, Jean-Philippe Goulet, Bartha Knoppers, Pavel Hamet, Claude Laberge, and on behalf of the CARTaGENE Project. 2013. <span>‘Cohort Profile of the <span>CARTaGENE</span> Study: <span>Quebec</span>’s Population-Based Biobank for Public Health and Personalized Genomics’</span>. <em>International Journal of Epidemiology</em> 42 (5): 1285–99. <a href="https://doi.org/10.1093/ije/dys160">https://doi.org/10.1093/ije/dys160</a>.
</div>
<div id="ref-bretz2016" class="csl-entry" role="listitem">
Bretz, Frank, Torsten Hothorn, and Peter Westfall. 2016. <em>Multiple <span>Comparisons Using R</span></em>. New York: <span>Chapman and Hall/CRC</span>. <a href="https://doi.org/10.1201/9781420010909">https://doi.org/10.1201/9781420010909</a>.
</div>
<div id="ref-bridges2025bb" class="csl-entry" role="listitem">
Bridges, Yasemin, Vinicius de Souza, Katherina G. Cortes, Melissa Haendel, Nomi L. Harris, Daniel R. Korn, Nikolaos M. Marinakis, et al. 2025. <span>‘Towards a Standard Benchmark for Phenotype-Driven Variant and Gene Prioritisation Algorithms: <span>PhEval</span>&nbsp;-&nbsp;<span>Phenotypic</span> Inference <span>Evaluation</span> Framework’</span>. <em>BMC Bioinformatics</em> 26 (1): 87. <a href="https://doi.org/10.1186/s12859-025-06105-4">https://doi.org/10.1186/s12859-025-06105-4</a>.
</div>
<div id="ref-chen2025b" class="csl-entry" role="listitem">
Chen, Ting-Huei, Chia-Jung Lee, Syue-Pu Chen, Shang-Jung Wu, and Cathy S J Fann. 2025. <span>‘<span>PNL</span>: A Software to Build Polygenic Risk Scores Using a Super Learner Approach Based on <span>PairNet</span>, a <span>Convolutional Neural Network</span>’</span>. <em>Bioinformatics</em> 41 (2): btaf071. <a href="https://doi.org/10.1093/bioinformatics/btaf071">https://doi.org/10.1093/bioinformatics/btaf071</a>.
</div>
<div id="ref-gnanaolivu2025bb" class="csl-entry" role="listitem">
Gnanaolivu, Rohan, Gavin Oliver, Garrett Jenkinson, Emily Blake, Wenan Chen, Nicholas Chia, Eric W. Klee, and Chen Wang. 2025. <span>‘A Clinical Knowledge Graph-Based Framework to Prioritize Candidate Genes for Facilitating Diagnosis of <span>Mendelian</span> Diseases and Rare Genetic Conditions’</span>. <em>BMC Bioinformatics</em> 26 (1): 82. <a href="https://doi.org/10.1186/s12859-025-06096-2">https://doi.org/10.1186/s12859-025-06096-2</a>.
</div>
<div id="ref-harris2025nrg" class="csl-entry" role="listitem">
Harris, Laura, Ellen M. McDonagh, Xiaolei Zhang, Katherine Fawcett, Amy Foreman, Petr Daneck, Panagiotis I. Sergouniotis, et al. 2025. <span>‘Genome-Wide Association Testing Beyond <span>SNPs</span>’</span>. <em>Nature Reviews Genetics</em> 26 (3): 156–70. <a href="https://doi.org/10.1038/s41576-024-00778-y">https://doi.org/10.1038/s41576-024-00778-y</a>.
</div>
<div id="ref-hayes2013gasagp" class="csl-entry" role="listitem">
Hayes, Ben. 2013. <span>‘Overview of <span>Statistical Methods</span> for <span>Genome-Wide Association Studies</span> (<span>GWAS</span>)’</span>. In <em>Genome-<span>Wide Association Studies</span> and <span>Genomic Prediction</span></em>, edited by Cedric Gondro, Julius van der Werf, and Ben Hayes, 149–69. Totowa, NJ: Humana Press. <a href="https://doi.org/10.1007/978-1-62703-447-0_6">https://doi.org/10.1007/978-1-62703-447-0_6</a>.
</div>
<div id="ref-kim2025b" class="csl-entry" role="listitem">
Kim, Na Yeon, and Seunggeun Lee. 2025. <span>‘<span>XPRS</span>: <span>A Tool</span> for <span>Interpretable</span> and <span>Explainable Polygenic Risk Score</span>’</span>. <em>Bioinformatics</em>, March, btaf143. <a href="https://doi.org/10.1093/bioinformatics/btaf143">https://doi.org/10.1093/bioinformatics/btaf143</a>.
</div>
<div id="ref-leeuw2015pcb" class="csl-entry" role="listitem">
Leeuw, Christiaan A. de, Joris M. Mooij, Tom Heskes, and Danielle Posthuma. 2015. <span>‘<span>MAGMA</span>: <span>Generalized Gene-Set Analysis</span> of <span>GWAS Data</span>’</span>. <em>PLOS Computational Biology</em> 11 (4): e1004219. <a href="https://doi.org/10.1371/journal.pcbi.1004219">https://doi.org/10.1371/journal.pcbi.1004219</a>.
</div>
<div id="ref-lin2024" class="csl-entry" role="listitem">
Lin, Jiecong, Ruibang Luo, and Luca Pinello. 2024. <span>‘<span>EPInformer</span>: A Scalable Deep Learning Framework for Gene Expression Prediction by Integrating Promoter-Enhancer Sequences with Multimodal Epigenomic Data’</span>. 1 August 2024. <a href="https://doi.org/10.1101/2024.08.01.606099">https://doi.org/10.1101/2024.08.01.606099</a>.
</div>
<div id="ref-menashe2008bg" class="csl-entry" role="listitem">
Menashe, Idan, Philip S Rosenberg, and Bingshu E Chen. 2008. <span>‘<span>PGA</span>: Power Calculator for Case-Control Genetic Association Analyses’</span>. <em>BMC Genetics</em> 9 (May): 36. <a href="https://doi.org/10.1186/1471-2156-9-36">https://doi.org/10.1186/1471-2156-9-36</a>.
</div>
<div id="ref-moore2019hh" class="csl-entry" role="listitem">
Moore, Camille M., Sean A. Jacobso, and Tasha E. Fingerlin. 2019. <span>‘Power and <span>Sample Size Calculations</span> for <span>Genetic Association Studies</span> in the <span>Presence</span> of <span>Genetic Model Mis-Specification</span>’</span>. <em>Human Heredity</em> 84 (6): 256–71. <a href="https://doi.org/10.1159/000508558">https://doi.org/10.1159/000508558</a>.
</div>
<div id="ref-mostafavi2023ng" class="csl-entry" role="listitem">
Mostafavi, Hakhamanesh, Jeffrey P. Spence, Sahin Naqvi, and Jonathan K. Pritchard. 2023. <span>‘Systematic Differences in Discovery of Genetic Effects on Gene Expression and Complex Traits’</span>. <em>Nature Genetics</em> 55 (11): 1866–75. <a href="https://doi.org/10.1038/s41588-023-01529-1">https://doi.org/10.1038/s41588-023-01529-1</a>.
</div>
<div id="ref-nolen2025b" class="csl-entry" role="listitem">
Nolen, Zachary J. 2025. <span>‘<span>PopGLen</span>—<span>A Snakemake</span> Pipeline for Performing Population Genomic Analyses Using Genotype Likelihood-Based Methods’</span>. <em>Bioinformatics</em>, March, btaf105. <a href="https://doi.org/10.1093/bioinformatics/btaf105">https://doi.org/10.1093/bioinformatics/btaf105</a>.
</div>
<div id="ref-pelletier2022" class="csl-entry" role="listitem">
Pelletier, Justin. 2022. <span>‘Évaluation de l’imputation Des Données Génétiques <span>Canadiennes-Françaises</span>’</span>, April. <a href="http://hdl.handle.net/1866/27582">http://hdl.handle.net/1866/27582</a>.
</div>
<div id="ref-perez-granado2022fg" class="csl-entry" role="listitem">
Pérez-Granado, Judith, Janet Piñero, and Laura I. Furlong. 2022. <span>‘Benchmarking Post-<span>GWAS</span> Analysis Tools in Major Depression: <span>Challenges</span> and Implications’</span>. <em>Frontiers in Genetics</em> 13 (October). <a href="https://doi.org/10.3389/fgene.2022.1006903">https://doi.org/10.3389/fgene.2022.1006903</a>.
</div>
<div id="ref-politi2023l" class="csl-entry" role="listitem">
Politi, Cristina, Stefanos Roumeliotis, Giovanni Tripepi, and Belinda Spoto. 2023. <span>‘Sample <span>Size Calculation</span> in <span>Genetic Association Studies</span>: <span>A Practical Approach</span>’</span>. <em>Life</em> 13 (1): 235. <a href="https://doi.org/10.3390/life13010235">https://doi.org/10.3390/life13010235</a>.
</div>
<div id="ref-schonherr2024" class="csl-entry" role="listitem">
Schönherr, Sebastian, and Lukas Forer. (2021) 2024. <span>‘Nf-Gwas - <span>A</span> Nextflow Pipeline to Perform <span>GWAS</span>.’</span> <a href="https://github.com/genepi/nf-gwas">https://github.com/genepi/nf-gwas</a>.
</div>
<div id="ref-sun2025bb" class="csl-entry" role="listitem">
Sun, Na, Qiang Han, Yu Wang, Mengtong Sun, Ziqing Sun, Hongpeng Sun, and Yueping Shen. 2025. <span>‘<span>BHCox</span>: <span>Bayesian</span> Heredity-Constrained <span>Cox</span> Proportional Hazards Models for Detecting Gene-Environment Interactions’</span>. <em>BMC Bioinformatics</em> 26 (1): 58. <a href="https://doi.org/10.1186/s12859-025-06077-5">https://doi.org/10.1186/s12859-025-06077-5</a>.
</div>
<div id="ref-uffelmann2021nrmp" class="csl-entry" role="listitem">
Uffelmann, Emil, Qin Qin Huang, Nchangwi Syntia Munung, Jantina de Vries, Yukinori Okada, Alicia R. Martin, Hilary C. Martin, Tuuli Lappalainen, and Danielle Posthuma. 2021. <span>‘Genome-Wide Association Studies’</span>. <em>Nature Reviews Methods Primers</em> 1 (1): 1–21. <a href="https://doi.org/10.1038/s43586-021-00056-9">https://doi.org/10.1038/s43586-021-00056-9</a>.
</div>
<div id="ref-wang2005nrg" class="csl-entry" role="listitem">
Wang, William Y. S., Bryan J. Barratt, David G. Clayton, and John A. Todd. 2005. <span>‘Genome-Wide Association Studies: Theoretical and Practical Concerns’</span>. <em>Nature Reviews Genetics</em> 6 (2): 109–18. <a href="https://doi.org/10.1038/nrg1522">https://doi.org/10.1038/nrg1522</a>.
</div>
<div id="ref-yang2024bib" class="csl-entry" role="listitem">
Yang, Zikun, Basilio Cieza, Dolly Reyes-Dumeyer, Rosa Montesinos, Marcio Soto-Añari, Nilton Custodio, and Giuseppe Tosto. 2024. <span>‘A Benchmark Study on Current <span>GWAS</span> Models in Admixed Populations’</span>. <em>Briefings in Bioinformatics</em> 25 (1): bbad437. <a href="https://doi.org/10.1093/bib/bbad437">https://doi.org/10.1093/bib/bbad437</a>.
</div>
<div id="ref-zhao2025bib" class="csl-entry" role="listitem">
Zhao, Gefei, and Binbin Lai. 2025. <span>‘<span>SC-VAR</span>: A Computational Tool for Interpreting Polygenic Disease Risks Using Single-Cell Epigenomic Data’</span>. <em>Briefings in Bioinformatics</em> 26 (2): bbaf123. <a href="https://doi.org/10.1093/bib/bbaf123">https://doi.org/10.1093/bib/bbaf123</a>.
</div>
<div id="ref-zhao2024gb" class="csl-entry" role="listitem">
Zhao, Zijie, Tim Gruenloh, Meiyi Yan, Yixuan Wu, Zhongxuan Sun, Jiacheng Miao, Yuchang Wu, Jie Song, and Qiongshi Lu. 2024. <span>‘Optimizing and Benchmarking Polygenic Risk Scores with <span>GWAS</span> Summary Statistics’</span>. <em>Genome Biology</em> 25 (1): 260. <a href="https://doi.org/10.1186/s13059-024-03400-w">https://doi.org/10.1186/s13059-024-03400-w</a>.
</div>
</div>
</section></section><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><hr>
<ol>
<li id="fn1"><p>Notably includes MHC, lactase region, and known inversions <code>8p23</code> and <code>17q21.31</code>. <strong>Haplotypes</strong>?<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2"><p><span data-fg="red" style="color: red;"><code>--king-cutoff 0.0884</code> corresponds to <code>PI_HAT = 0.125</code>, see details <a href="http://biostars.org/p/434832/#434898">here</a></span><a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn3"><p><code>DMOTSCORE_mod</code> represents the measurement of bone mineral density at the calcaneus. While no universally accepted threshold exists for the diagnosis of osteopenia or osteoporosis, we adopt the conventional cut-off of <code>-1.5</code>.<a href="#fnref3" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn4"><p>For both <code>FXMOF_POST</code> and <code>FXALL_POST</code>, the dates of occurrence and censoring dates are available in <code>FXALL_POSTDATE</code> and <code>FXMOF_POSTDATE</code> respectively.<a href="#fnref4" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn5"><p>Use of dnaseq <code>Genpipes</code> pipelines, run on Compute Canada Clusters, version 3.1.5.<a href="#fnref5" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn6"><p>Avoid using <code>BiomarT</code> package a tall costs: it’s not anymore maintained, and leveraged out-of-date <code>Hg37</code> Human Genome Build. Report to <a href="https://www.linkedin.com/posts/%F0%9F%8E%AF-ming-tommy-tang-40650014_genome-builds-matter-avoid-costly-mistakes-activity-7293270985930141698-ZFkH">Genome Builds Versions</a> for details.<a href="#fnref6" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn7"><p>All curated VCF files per chromosome are stored in <code>./data/genotypages/</code> (symbolly linked with <code>/mnt/projects_tn01/Cartagene/genotypage/imputation/imputation_merged</code>).<a href="#fnref7" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn8"><p><code>nohup</code> option combined with <code>&amp;</code> character enables running variants extraction in the background, even with terminal disrupting and closing.<a href="#fnref8" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn9"><p><code>dosage=HDS</code>, for Hard Dosage, provides the instruction describing genotype uncertainties for enhanced statistical power. It’s particularly useful when working as here with <strong>imputed genotypes</strong>, where part of the SNPs were inferred using reference panels.<a href="#fnref9" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn10"><p>The <code>hide-covar</code> option runs GLM, but doesn’t output covariate results. Besides, note that <code>--covar</code> passes <em>quantitative covariates</em> (e.g., age, PCs), while <code>--ide-covar</code> passes <em>categorical covariates</em><a href="#fnref10" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn11"><p><strong>Note that a VIF factor above 10 is considered problematic as it can significantly distort regression estimates.</strong> Discussion and visualisations on <a href="https://www.linkedin.com/feed/update/urn:li:activity:7308463515432833026/"><em>VIF influence</em></a>. Briefly, VIF quantifies the degree of similarity across predictor variables, as the GLM-family usually assumes independence between explanatory variables. When VIF is significant, independence assumption is discarded, which might be explained by the fact that preprocessing in <a href="#sec-preprocessing-redundancy" class="quarto-xref"><span>Section 1.3.2.2</span></a> for removing correlated SNps and individuals are not used in subsequent GWAS differential downstream analyses.<a href="#fnref11" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn12"><p>More customisation available in <a href="https://r-graph-gallery.com/101_Manhattan_plot.html">Manhattan plot in R: a review</a><a href="#fnref12" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn13"><p><a href="https://varianceexplained.org/statistics/interpreting-pvalue-histogram/">How to interpret a <span class="math inline">\(p\)</span>-value histogram</a><a href="#fnref13" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn14"><p><a href="https://danielroelfs.com/blog/how-i-make-qq-plots-using-ggplot/">How I Make QQ Plots Using <code>ggplot2</code>?</a><a href="#fnref14" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn15"><p>Note that relying on HPO terms, for example, <code>HP:0004322 – Osteopenia</code>, would alleviate the need for standardising phenotype labels, using universally acknowledged scientific terms instead.<a href="#fnref15" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/arnauddroitlab\.github\.io\/cartagene-gwas\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      const annoteTargets = window.document.querySelectorAll('.code-annotation-anchor');
      for (let i=0; i<annoteTargets.length; i++) {
        const annoteTarget = annoteTargets[i];
        const targetCell = annoteTarget.getAttribute("data-target-cell");
        const targetAnnotation = annoteTarget.getAttribute("data-target-annotation");
        const contentFn = () => {
          const content = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          if (content) {
            const tipContent = content.cloneNode(true);
            tipContent.classList.add("code-annotation-tip-content");
            return tipContent.outerHTML;
          }
        }
        const config = {
          allowHTML: true,
          content: contentFn,
          onShow: (instance) => {
            selectCodeLines(instance.reference);
            instance.reference.classList.add('code-annotation-active');
            window.tippy.hideAll();
          },
          onHide: (instance) => {
            unselectCodeLines();
            instance.reference.classList.remove('code-annotation-active');
          },
          maxWidth: 300,
          delay: [50, 0],
          duration: [200, 0],
          offset: [5, 10],
          arrow: true,
          appendTo: function(el) {
            return el.parentElement.parentElement.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'quarto',
          placement: 'right',
          popperOptions: {
            modifiers: [
            {
              name: 'flip',
              options: {
                flipVariations: false, // true by default
                allowedAutoPlacements: ['right'],
                fallbackPlacements: ['right', 'top', 'top-start', 'top-end', 'bottom', 'bottom-start', 'bottom-end', 'left'],
              },
            },
            {
              name: 'preventOverflow',
              options: {
                mainAxis: false,
                altAxis: false
              }
            }
            ]        
          }      
        };
        window.tippy(annoteTarget, config); 
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><nav class="page-navigation column-body"><div class="nav-page nav-page-previous">
      <a href="./index.html" class="pagination-link" aria-label="Welcome">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Welcome</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./tools_installation.html" class="pagination-link" aria-label="Tools installation">
        <span class="nav-page-text"><span class="chapter-number">A</span>&nbsp; <span class="chapter-title">Tools installation</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
<p>Written by Bastien CHASSAGNOL and Marie-Pier SCOTT-KELLY, with <a href="https://quarto.org/">Quarto</a>.</p>
<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/ArnaudDroitLab/cartagene-gwas/edit/master/CARTaGENE-GWAS-Report.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ArnaudDroitLab/cartagene-gwas/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/ArnaudDroitLab/cartagene-gwas/blob/master/CARTaGENE-GWAS-Report.qmd" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
<li class="nav-item compact">
    <a class="nav-link active" href="https://github.com/ArnaudDroitLab/cartagene-gwas" aria-current="page">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
</ul>
</div>
  </div>
</footer><script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>


<script src="site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>